---
title: "RhizMicrobiome_GenusLevel"
author: "Sara Colom"
date: "2/8/2020"
output: 
  html_document:
    toc: true
    toc_float: true
    highlight: tango
    code_folding: hide
    self_contained: yes 
---

# R version

```{r}
version
```


## Sample sizes

### Table sample number by species and treatment


| Species          | Treatment       | N      |
|--------------    |-------------    |----    |
| I. purpurea      | Alone           | 27     |
| I. purpurea      | Competition     | 78     |
| I. hederacea     | Competition     | 78     |


### Table number of maternal line per species

| Species          | Number of ML     |
|--------------    |--------------    |
| I. purpurea      |      10          |
| I. hederacea     |       5          |




### Load Libraries
```{r load_libs, include=T,warning=F,comment=F,message=F}
library(phyloseq)
library(ggplot2)
library(ape)
library(vegan)
library(plyr)
library(dplyr)
library(scales)
library(grid)
library(reshape2)
#library(DESeq2)
library(pegas)
library(pgirmess)
library(multcomp)
library(ggpubr)
library(ggcorrplot)
library(RColorBrewer)
library(plotly)
library(ggthemes)
library(corrplot)
library(Hmisc)
library(lmerTest)


source("miSeq.R")

# Aesthetics
Tx<-theme(axis.text.y = element_text(size=20),
          axis.title.y = element_text(size=20))+
          theme(axis.text.x = element_text(vjust = 1, hjust=1,angle=0,size=20),
          axis.title.x = element_text(angle=0,size=20),
          plot.title=element_text(size=25,hjust=0))

# Margins
Margin<-theme(
  panel.background = element_rect(fill = "white"),
  plot.margin = margin(2, 2, 2, 2, "cm"),
  plot.background = element_rect(
    fill = "white",
    colour = "black",
    size = 1
  )
)

# Aesthetics
Tx2<-theme(axis.text.y = element_text(size=20),
          axis.title.y = element_text(size=25))+
          theme(axis.text.x = element_text(vjust = 1, hjust=1,size=25),
          axis.title.x = element_text(size=25),
          plot.title=element_text(size=25,hjust=0))

GoldGrey=c("#F1CE63","#79706E")
GreenBlue=c("#59A14F","#4E79A7")

```



## Sequence data analysis
```{r Import Data,warning=F,comment=F,message=F}
### loading mothur output with FWDB+silva taxonomy and sample metadata. 
### Experiments run in 

sharedfile <- "../DataSets/stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.shared"
taxfile <- "../DataSets/stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.taxonomy"

mothurdata <- import_mothur(mothur_shared_file = sharedfile,mothur_constaxonomy_file = taxfile)

sampledata= read.csv('../DataSets/MetaDataTest.csv')

SAMPLE=sampledata
row.names(SAMPLE)=SAMPLE$Sample_ID

SAMPLE=subset(SAMPLE,SAMPLE$TRT=="Alone"|SAMPLE$TRT=="Inter")
SAMPLE=sample_data(SAMPLE)

### create phyloseq object
physeq.all = merge_phyloseq(mothurdata, SAMPLE) # Modified version worked


### We need to change the taxonomy names: when using the fwdb taxonomy we need to add different headers after removing the last column, which contains no information except for 1 Verrucomicrobia taxon

#   tax_table(physeq.all) <- tax_table(physeq.all)[,-7] # this removes the final column
colnames(tax_table(physeq.all))<-c("Kingdom","Phylum","Class","Order","Family","Genus")


#   tax_table(physeq.all) <-cbind(tax_table(physeq.all),row.names(tax_table(physeq.all)))

### removing non-bacterial reads (was already done in mothur, but just to be safe after merging taxonomies)
#   physeq.all <- subset_taxa(physeq.all, Kingdom == "Bacteria")

```



```{r, aggregate data by genus}

## Aggregate at the genus level, then saving ea componant externally to reload and save time.

        # physeq_genus <- physeq.all %>%
         #tax_glom(taxrank = "Genus") 

        # otu=data.frame(otu_table(physeq_genus))
        
        # meta=data.frame(sample_data(physeq_genus))
        # tax=data.frame(tax_table(physeq_genus))
        
        # write.csv(x = otu,file = "otu_table.csv",row.names = TRUE)
        # write.csv(x = tax,file = "tax_table.csv")# 
        # write.csv(x = meta,file = "sample_table.csv",row.names = F)


otu = read.csv("../DataSets/otu_table.csv",row.names = 1)

colnames(otu)=gsub('X',"",colnames(otu))

otu=otu_table(otu,taxa_are_rows = TRUE)


tax=read.csv('../DataSets/tax_table.csv',row.names=1)
taxRows=row.names(tax)
taxCols=colnames(tax)
tax=tax_table(as.matrix(tax))

#row.names(tax)=taxRows
#colnames(tax)=taxCols


meta = read.csv('../DataSets/MetaDataTest.csv')
namesKeep<-colnames(otu)

row.names(meta)=meta$Sample_ID

meta=meta[which(meta$Sample_ID%in%namesKeep),]
meta=sample_data(meta)
   
physeq_genus=merge_phyloseq(otu,tax,meta)

```



```{r Proteobacteria phylum change to class,warning=F,comment=F,message=F}
### ADD THE PROTEOBACTERIA CLASSES TO THE PHYLA NAME FIELD IN PHYLOSEQ OBJECT TAXONOMY 

phy <- data.frame(tax_table(physeq_genus))
Phylum <- as.character(phy$Phylum)
Class <- as.character(phy$Class)
for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Proteobacteria"){
    if (Class[i] == "unclassified"){
      Phylum[i] <- Phylum[i]       
    } else {
      Phylum[i] <- Class[i]
    }
  } 
}


Phylum<-as.character(phy$Phylum)
Class=as.character(phy$Class)
phy$Phylum=Phylum
t <- tax_table(as.matrix(phy))

```


```{r read in phylum data, include=FALSE,warning=F,comment=F,message=F}
SAMPLE=sample_data(physeq_genus)
SAMPLE$Species=ifelse(grepl("Ip",SAMPLE$ML),"Ip","Ihed")
physeq.all2 = merge_phyloseq(physeq_genus, SAMPLE) 
```

## Sequence depth and pruning

```{r data scaling}
physeq=(physeq.all2)
physeq <- prune_taxa(taxa_sums(physeq) > 0, physeq) # Remove taxa with no counts

#check number of reads in each sample, differences in count are in part due differet numbers of chlorophyl reads depending on time of experiment

# Make a data frame with a column for the read counts of each sample
sample_sum_df <- data.frame(sum = sample_sums(physeq))

# Histogram of sample read counts
ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "#59A14F", binwidth = 2500) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank())+
  theme_classic()


# Scales reads to smallest library size 
source("https://raw.githubusercontent.com/michberr/MicrobeMiseq/master/R/miseqR.R")
#physeq.scale <- scale_reads(physeq, min(sample_sums(physeq)))

##### Normalization #######

# Scales reads by 
# 1) taking proportions,
# 2) multiplying by a given library size of n
# 3) rounding down
physeq1=prune_samples(sample_sums(physeq) >20000, physeq)
sample_sum_df1 <- data.frame(sum = sample_sums(physeq1))

# Histogram of sample read counts
ggplot(sample_sum_df1, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "#59A14F", binwidth = 2500) +
  ggtitle("Distribution of sample sequencing depth after pruning") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank())+
  theme_classic()


n=min(sample_sums(physeq1))
  physeq.scale <-
    transform_sample_counts(physeq1, function(x) {
      (n * x/sum(x))
    })
physeq.scale <- prune_taxa(taxa_sums(physeq.scale) > 0, physeq.scale)


physeq1

```

```{r PCoA Treatment and Species comparisoin Bray Curtis Estimate,warning=F,comment=F,message=F}
# Betadiv pcoa Bray-Curtis DNA only 
physeq.pcoa <-
  ordinate(
    physeq = physeq.scale,
    method = "PCoA",
    distance = "bray"
  )

physeq.pcoa.vectors <- data.frame(physeq.pcoa$vectors[, 1:4])
physeq.pcoa.vectors$Duplicates <- row.names(physeq.pcoa.vectors)
SampData=data.frame(sample_data(physeq))
colnames(SampData)[1]="Duplicates"

SampData=subset(SampData,SampData$TRT=="Inter"|SampData$TRT=="Alone")

physeq.pcoa.df <- droplevels(merge(physeq.pcoa.vectors,SampData,by="Duplicates"))
bray_values <- physeq.pcoa$values
bray_rel_eigens <- bray_values$Relative_eig
bray_rel_eigen1 <- bray_rel_eigens[1]
bray_rel_eigen1_percent <- round(bray_rel_eigen1 * 100, digits = 1)
bray_rel_eigen2 <- bray_rel_eigens[2]
bray_rel_eigen2_percent <- round(bray_rel_eigen2 * 100, digits = 1)
bray_rel_eigen3 <- bray_rel_eigens[3]
bray_rel_eigen3_percent <- round(bray_rel_eigen3 * 100, digits = 1)
bray_rel_eigen4 <- bray_rel_eigens[4]
bray_rel_eigen4_percent <- round(bray_rel_eigen4 * 100, digits = 1)
bray_rel_eigen5 <- bray_rel_eigens[5]
bray_rel_eigen5_percent <- round(bray_rel_eigen5 * 100, digits = 1)
bray_axis1 <- paste("PCoA 1:",bray_rel_eigen1_percent,"%")
bray_axis2 <- paste("PCoA 2:",bray_rel_eigen2_percent,"%")
bray_axis3 <- paste("PCoA 3:",bray_rel_eigen3_percent,"%")
bray_axis4 <- paste("PCoA 4:",bray_rel_eigen4_percent,"%")
PCoA_title <- paste("Bray-Curtis,",ntaxa(physeq.scale),"OTUs")


pcoa_exp_trt <- ggplot(physeq.pcoa.df, aes(Axis.1, Axis.2, color = TRT,fill = TRT)) +
  xlab(bray_axis1) + 
  ylab(bray_axis2) +
  geom_point(alpha=0.9) + 
  theme_classic() +
  scale_fill_manual(values = c("#00B050","grey"),"Treatment", labels = c("Alone", "Competition"))+
  scale_color_manual(values = c("#00B050","grey"),"Treatment", labels = c("Alone", "Competition"))+
  stat_ellipse(geom = "polygon", alpha = 1/6, aes(fill =TRT))+
  Tx+
  theme(axis.text.x = element_text(angle=45),axis.text=element_text(color="black",size=15,vjust = 0.5,hjust=1))
#  scale_fill_discrete(name = "Treatment", labels = c("Alone", "Competition"))+
 # scale_color_discrete(name = "Treatment", labels = c("Alone", "Competition"))



pcoa_exp_trtA <- ggplot(physeq.pcoa.df, aes(Axis.3, Axis.4, color = TRT,fill = TRT)) +
  xlab(bray_axis3) + 
  ylab(bray_axis4) + 
  geom_point(alpha=0.9) + 
  theme_classic() +
  scale_fill_manual(values = GoldGrey,"Treatment", labels = c("Alone", "Competition"))+
  scale_color_manual(values = GoldGrey,"Treatment", labels = c("Alone", "Competition"))+
  stat_ellipse(geom = "polygon", alpha = 1/6, aes(fill =TRT))+
  Tx


pcoa_exp_sp <- ggplot(physeq.pcoa.df, aes(Axis.1, Axis.2, color = Species,fill = Species)) +
  xlab("") + 
  ylab(bray_axis2) +
  geom_point(alpha=0.9) + 
  theme_classic() +
 scale_fill_manual(values = GreenBlue,"Species", labels = c("I. hederacea", "I. purpurea"))+
  scale_color_manual(values = GreenBlue,"Species", labels = c("I. hederacea", "I. purpurea"))+
   stat_ellipse(geom = "polygon", alpha = 1/6, aes(fill =Species))+
  Tx


pcoa_exp_spA <- ggplot(physeq.pcoa.df, aes(Axis.3, Axis.4, color = Species,fill = Species)) +
  xlab("") + 
  ylab(bray_axis4) +
  geom_point(alpha=0.9) + 
  theme_classic()+
  scale_fill_manual(values = GreenBlue,"Species", labels = c("I. hederacea", "I. purpurea"))+
  scale_color_manual(values = GreenBlue,"Species", labels = c("I. hederacea", "I. purpurea"))+
  stat_ellipse(geom = "polygon", alpha = 1/6, aes(fill =Species))+
  Tx

A=ggarrange(pcoa_exp_trt,pcoa_exp_trtA,common.legend = T,labels=c("C","D"),font.label = list(size = 15, color = "black", face =
  "plain"),hjust = -8,vjust=1)
B=ggarrange(pcoa_exp_sp,pcoa_exp_spA,common.legend = T,labels=c("A","B"),font.label = list(size = 15, color = "black", face =
  "plain"),hjust = -8,vjust=1)

MainFig=ggarrange(B,A,nrow=2)

annotate_figure(MainFig,
                top = text_grob(PCoA_title, face = "plain", size = 25,lineheight = 2)
)

```



```{r 3D graphic,warning=F,comment=F,message=F}

physeq.pcoa.df$TrtSp=paste(physeq.pcoa.df$TRT,physeq.pcoa.df$Species,sep="")

p <- plot_ly(physeq.pcoa.df, x = ~Axis.1, y = ~Axis.2, z = ~Axis.3, color = ~TrtSp) %>%
  add_markers() %>%
  layout(scene = list(xaxis = list(title = bray_axis1),
                     yaxis = list(title = bray_axis2),
                     zaxis = list(title = bray_axis3)))


p1 <- plot_ly(physeq.pcoa.df, x = ~Axis.1, y = ~Axis.2, z = ~Axis.4, color = ~Species) %>%
  add_markers() %>%
  layout(scene = list(xaxis = list(title = bray_axis1),
                     yaxis = list(title = bray_axis2),
                     zaxis = list(title = bray_axis4)))

p3 <- plot_ly(physeq.pcoa.df, x = ~Axis.1, y = ~Axis.2, z = ~Axis.4, color = ~Species) %>%
  add_markers() %>%
  layout(scene = list(xaxis = list(title = bray_axis1),
                     yaxis = list(title = bray_axis3),
                     zaxis = list(title = bray_axis4)))

```

```{r adonis experimental data when all data combined,warning=F,comment=F,message=F}
## All experiment, add variable treatment, then species
sampledf <- data.frame(sample_data(physeq.scale)) %>% 
  select(-Comment)

## Subsample by species 

physeq.purp <- subset_samples(physeq.scale, Species == "Ip")

treatment <- sampledf %>% 
  filter(Species == "Ip") %>% 
  pull(TRT)

block <- sampledf %>% 
  filter(Species == "Ip") %>% 
  pull(Block)

## Subsample within competition

physeq.comp = subset_samples(physeq.scale, TRT=="Inter")

# Adonis test for treatment within I.purpurea

adonis(otu_table(physeq.purp) %>% t ~ treatment + block, method = "bray")



# Adonis test for species 

```


```{r AMOVA}

# # # # # # # # # # # # # # # # # # # # # 
#             ACROSS SAMPLES

physeq.bray <- phyloseq::distance(physeq = physeq.scale, method = "bray")


amova(physeq.bray ~ TRT, data = sampledf) # Across all samples
amova(physeq.bray ~ Species, data = sampledf) # Across all samples

# # # # # # # # # # # # # # # # # # # # # 
# Subsample for within I. purpurea only
# # # # # # # # # # # # # # # # # # # # # 

physeq.Purp <- subset_samples(physeq.scale, Species == "Ip")
sampledf.Purp<- data.frame(sample_data(physeq.Purp))

# Calculate bray curtis for summer samples only
physeq.Purp.bray <- phyloseq::distance(physeq = physeq.Purp, method = "bray")
amova(physeq.Purp.bray ~ TRT, data = sampledf.Purp) # Across all samples


# # # # # # # # # # # # # # # # # # # # # 
# Subsample for within competition
# # # # # # # # # # # # # # # # # # # # # 

physeq.Comp <- subset_samples(physeq.scale, TRT=="Inter")
sampledf.Comp<- data.frame(sample_data(physeq.Comp))

# Calculate bray curtis for summer samples only
physeq.Comp.bray <- phyloseq::distance(physeq = physeq.Comp, method = "bray")
amova(physeq.Comp.bray ~ Species, data = sampledf.Comp) # marginal species differences

```


## Test for differently abundant OTUs

### Visualize the different groups at the family level
```{r}
# Check for 'core' microbiome members as the phylum level which taxa are present
# at 1 % overall abundance and at least 75% of samples 

PhylumGlom <- tax_glom(physeq.Purp,taxrank="Phylum")
FamGlom <- tax_glom(physeq.Purp,taxrank="Family")
ClassGlom <- tax_glom(physeq.Purp,taxrank="Class")


coreTaxa <- filter_taxa(PhylumGlom, function(x) sum(x > 1) > (0.75 * length(x)), TRUE)
coreTaxaFamily <- filter_taxa(FamGlom, function(x) sum(x > 1) > (0.75 * length(x)), TRUE)

## Plot at phylum level


physeq1_phylumAlone <- coreTaxa  %>%
  subset_samples(TRT=="Alone")%>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt()       

# Melt to long format
physeq1_phylumAlone = physeq1_phylumAlone[order(physeq1_phylumAlone$Phylum),] 

physeq1_phylumComp <- coreTaxa  %>%
  subset_samples(TRT!="Alone")%>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt()       

physeq1_phylum<-rbind(physeq1_phylumAlone,physeq1_phylumComp)

#colnames(physeq1_phylum)[13]="Phylum"

# Sort data frame alphabetically by phylum
# Set colors for plotting
phylum_colors <- c(
  "#CBD588", "#5F7FC7", "orange","#DA5724", "#508578", "#CD9BCD",
   "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861"
)


#colnames(physeq1_phylum)[which(names(physeq1_phylum)%in%"Rank2")]="Phylum"

# Plot 
ggplot(physeq1_phylum, aes(x = TRT, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity",position='dodge')+
  #scale_fill_manual(values = phylum_colors)+
  theme(axis.title.x = element_blank()) + 
  ylab("Relative Abundance (Phyla > 75 %) \n") +
  ggtitle("Phylum Composition  \n Bacterial Communities by Sampling per Treatment") +
  theme_classic()+
  ylab("Relative Abundance")

```


## Testing if different at the phylum level

```{r}

PhlyaList <- as.list(unique(physeq1_phylum$Phylum))

KrusMicrTest <- function(Microbe){
  Result <- kruskal.test(Abundance ~ TRT, data = physeq1_phylum[which(physeq1_phylum$Phylum == Microbe),]) 

ResDf <- data.frame("ChiSq" = Result$statistic,"Pval" = Result$p.value,"Phylum" = paste(Microbe))  
  return(ResDf)  
}

AllResults <- lapply(PhlyaList,KrusMicrTest)
KruskalRes <- do.call('rbind',AllResults)
```


## Family level
```{r}

# Family level
physeq1_FamilyAlone <- coreTaxaFamily  %>%
  subset_samples(TRT == "Alone")%>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt()                                      # Melt to long format

physeq1_FamilyComp <- coreTaxaFamily  %>%
  subset_samples(TRT != "Alone")%>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt()       

physeq1_Family<-rbind(physeq1_FamilyAlone,physeq1_FamilyComp)


FamList<-as.list(unique(physeq1_Family$Family))

KrusMicrTest<-function(Microbe){
  Result = kruskal.test(Abundance ~ TRT, data = physeq1_Family[which(physeq1_Family$Family == Microbe),]) 

ResDf <- data.frame("ChiSq" = Result$statistic,"Pval" = Result$p.value,"Family" = paste(Microbe))  
  return(ResDf)  
}

AllResultsFamily <- lapply(FamList,KrusMicrTest)
KruskalResFamily <- do.call('rbind',AllResultsFamily)

KruskalResFamily %>%
  filter(Pval<0.055)

# Do false discovery rate for mulitple corrections

KruskalResFamily$AdjPval <- p.adjust(KruskalResFamily$Pval, method = "fdr", n = 213)
```


## Alpha diversity

```{r alpha diversity,warning=F,comment=F,message=F}
hist(sample_sums(physeq1))

# Remove sample with less than 20K reads, this looks WAAAY off
# Rarify first
ps.rarefied <- rarefy_even_depth(physeq1,rngseed = 1,sample.size=min(sample_sums(physeq1)),replace = F)

plot_richness(ps.rarefied,x ="TRT",measures = c("Observed","Shannon")) +
  geom_boxplot() +
  theme_classic()
```


## Alpha Diversity

#### Note: Measures of alpha-diversity 

_*Inverse Simpson*_ it is an indication of the richness in a community with uniform evenness that would have the same level of diversity. So while measures such as the Shannon index are somewhat abstract, the inverse of the Simpson index has some biological interpretation. Other advantages of the Simpson-based metrics are that they do not tend to be as affected by sampling effort as the Shannon index.
   
- _*Species richness*_ is simply a count of species, and it does not take into account the abundances of the species or their relative abundance distributions.
   
- _*Simpson's Diversity Index*_ is a measure of diversity which takes into account the number of species present, as well as the relative abundance of each species. As species richness and evenness increase,so diversity increases.



```{r Richness and Inverse Simpson,warning=F,comment=F,message=F}
# Initialize matrices to store richness and Inverse Simpson estimates

trials = 100

richness <- matrix(nrow = 173, ncol = trials)
row.names(richness) <- sample_names(physeq1)

evenness <- matrix(nrow = 173, ncol = trials)
row.names(evenness) <- sample_names(physeq1)

simpson <- matrix(nrow = 173, ncol = trials)
row.names( simpson) <- sample_names(physeq1)

InvSimp <- matrix(nrow = 173, ncol = trials)
row.names(InvSimp) <- sample_names(physeq1)

shannon <- matrix(nrow = 173, ncol = trials)
row.names(shannon) <- sample_names(physeq1)


### I commented out the subsampling to save time. The matrices produced were saved and then reoponed from their tab delmited formats into corresponding matrices.

set.seed(3)

r <- rarefy_even_depth(physeq1, sample.size = n, verbose = FALSE, replace = TRUE)
         
 ## Calculate richness
rich <- as.numeric(as.matrix(estimate_richness(r, measures = "Observed")))

## Calculate Inverse Simpson

simp <- as.numeric(as.matrix(estimate_richness(r, measures = "InvSimpson")))

sim<- as.numeric(as.matrix(estimate_richness(r, measures = "Simpson")))

s <- rarefy_even_depth(physeq1, sample.size = n, verbose = FALSE, replace = TRUE)
shan <- as.numeric(as.matrix(estimate_richness(r, measures = "Shannon")))

InvSimp = simp

```




```{r}
# Create a new dataframe to hold the means and standard deviations of richness estimates
length(rich)

Sample_ID <- sample_names(physeq1)
Block <- sample_data(physeq1)$Block
Species <- sample_data(physeq1)$Species
TRT <- sample_data(physeq1)$TRT
Combos <- sample_data(physeq1)$Combos
ML <- sample_data(physeq1)$ML

alpha <- data.frame(Sample_ID, ML, Block, TRT, Species, Combos, rich, InvSimp, sim, shan)

alpha$even <- alpha$shan/alpha$rich

# DO VIOLIN PLOT HERE!!!


p <- ggplot(alpha %>%
              filter(TRT ==
                       "Inter"), aes(x = Species, y = rich)) +
  geom_violin(trim = FALSE, aes(fill = Species), alpha = 0.3) + 
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize=1,aes(color = Species, fill = Species))+
  theme_classic()+
  scale_color_brewer(palette = "Paired")+
  scale_fill_brewer(palette = "Paired")+
  ggtitle("Mean Species Richness")


q <- ggplot(alpha %>%
              filter(TRT == "Inter"), aes(x = Species, y = InvSimp)) +
  geom_violin(trim = FALSE, aes(fill = Species),alpha = 0.3)+ 
  geom_dotplot(binaxis ='y', stackdir = 'center', dotsize = 1,aes(color = Species, fill = Species))+
  theme_classic()+
  scale_color_brewer(palette = "Paired")+
  scale_fill_brewer(palette = "Paired")+
  ggtitle("Mean Species Inverse Simpson")+
  ylab("")


t <- ggplot(alpha %>% 
              filter(TRT == "Inter"), aes(x = Species, y = sim)) +
  geom_violin(trim = FALSE,aes(fill = Species),alpha = 0.3)+ 
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize=1,aes(color=Species,fill=Species))+
  theme_classic()+
  scale_color_brewer(palette = "Paired")+
  scale_fill_brewer(palette = "Paired")+
  ggtitle("Mean Simpson")+
  ylab("")

v <- ggplot(alpha %>% 
              filter(TRT == "Inter"), aes(x = Species, y = even)) +
  geom_violin(trim = FALSE, aes(fill = Species), alpha = 0.3)+ 
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1,aes(color = Species, fill = Species))+
  theme_classic()+
  scale_color_brewer(palette = "Paired")+
  scale_fill_brewer(palette = "Paired")+
  ggtitle("Mean Evenness")+
  ylab("")



ggarrange(p,q,t,v, common.legend = T, ncol = 2,nrow = 2)



      ### Treatment

pT <- ggplot(alpha %>%
               filter(Species == "Ip"), aes(x = TRT, y = rich)) +
  geom_violin(trim = FALSE, aes(fill = TRT), alpha = 0.3)+ 
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize=1,aes(color = TRT, fill = TRT))+
  theme_classic()+
  scale_color_manual(values = c("#B2DF8A","#fee090"))+
  scale_fill_manual(values = c("#B2DF8A","#fee090"))+
  ggtitle("Mean Treatment Richness")


qT <- ggplot(alpha %>% 
               filter(Species == "Ip"), aes(x = TRT, y = InvSimp)) +
  geom_violin(trim = FALSE, aes(fill = TRT),alpha = 0.3)+ 
  geom_dotplot(binaxis ='y', stackdir ='center', dotsize = 1,aes(color = TRT, fill = TRT))+
  theme_classic()+
  scale_color_manual(values = c("#B2DF8A","#fee090"))+
  scale_fill_manual(values = c("#B2DF8A","#fee090"))+
  ggtitle("Mean Treatment Inverse Simpson")+
  ylab(" ")

ggarrange(pT, qT, common.legend = T)

```

## Test for differences


```{r ANOVAS on alpha diversities,warning=F,comment=F,message=F}
# first check for normality. To test for normalcy statistically, we can run the Shapiro-Wilk test of normality.

shapiro.test(alpha$rich) # Not normal
shapiro.test(alpha$even) # Normal
shapiro.test(alpha$sim) # Normal

histogram(alpha$rich,xlab = "Richness") # Note richness though, not normal, looks close to normal.

histogram(alpha$InvSimp,xlab = "Inverse-Simpson")
histogram(alpha$sim,xlab = "Simpson")


#   Run the ANOVA and save it as an object

# Test for species within competition
aov.Richness = aov(rich~Species+Block,alpha %>% filter(TRT == "Inter"))
aov.simpsonInv = aov(InvSimp~Species+Block,alpha %>% filter(TRT == "Inter"))
aov.simpson = aov(sim~Species+Block,alpha %>% filter(TRT == "Inter"))
aov.evenness = aov(even~Species+Block,alpha %>% filter(TRT == "Inter"))
aov.shannon = aov(shan~Species+Block,alpha %>% filter(TRT == "Inter"))

#Call for the summary of that ANOVA, which will include P-values
summary(aov.Richness)
summary(aov.simpsonInv)
summary(aov.simpson)
summary(aov.evenness)
summary(aov.shannon)

# No significant differences


# Test for treatment within I. purpurea
alpha.purp <- alpha %>% filter(Species == "Ip")
alpha.purp$TRT <- as.factor(alpha.purp$TRT)
alpha.purp$Block <- as.factor(alpha.purp$Block)

aov.Richness <- lm(rich ~ TRT + Block, alpha.purp)
aov.simpsonInv <- lm(InvSimp ~ TRT + Block,alpha.purp)
aov.simpson <- lm(sim ~ TRT + Block,alpha.purp)
aov.evenness <- lm(even ~ TRT + Block,alpha.purp)
aov.shannon <- lm(shan ~ TRT + Block,alpha.purp)


#Call for the summary of that ANOVA, which will include P-values
anova(aov.Richness)
anova(aov.simpsonInv)
anova(aov.simpson)
anova(aov.evenness)
anova(aov.shannon)

# No significant differences



# Boxplot
ggplot(alpha %>% filter(Species == "Ip"),aes(TRT,rich))+
  geom_boxplot()+
  theme_bw()+
  ylab("Richness")+
  xlab("")+
  scale_x_discrete(labels = c("Alone","Competition"))+
  theme(axis.text.x = element_text(vjust = -1),axis.text = element_text(size = 20),axis.title = element_text(size=30),axis.title.x = element_text(vjust = -1))+
  Margin

ggplot(alpha %>% filter(Species == "Ip"),aes(TRT, even))+
  geom_boxplot()+
  theme_bw()+
  ylab("Evenness")+
  xlab("")+
  scale_x_discrete(labels = c("Alone", "Competition"))+
  theme(axis.text.x = element_text(vjust = -1),axis.text=element_text(size=20),axis.title=element_text(size = 30),axis.title.x = element_text(vjust = -1))+
  Margin

ggplot(alpha %>% filter(Species == "Ip"),aes(TRT,sim))+
  geom_boxplot()+
  theme_bw()+
  ylab("Simpson")+
  xlab("")+
  scale_x_discrete(labels = c("Alone","Competition"))+
  theme(axis.text.x = element_text(vjust = -1),axis.text=element_text(size=20),axis.title = element_text(size = 30),axis.title.x = element_text(vjust = -1))+
  Margin

ggplot(alpha %>% filter(Species == "Ip"),aes(TRT,InvSimp))+
  geom_boxplot()+
  theme_bw()+
  ylab("Inverse Simpson")+
  xlab("")+
  scale_x_discrete(labels = c("Alone","Competition"))+
  theme(axis.text.x = element_text(vjust = -1),axis.text=element_text(size = 20),axis.title = element_text(size = 30),axis.title.x = element_text(vjust = -1))+
  Margin

```

# Linear mixed models
```{r}
# Test for treatment within I. purpurea
alpha$Comp=sub(".*\\-","",alpha$Combos)

RichLmm = lmer(rich ~ TRT + Block + (1|Block:ML), alpha %>% filter(Species == "Ip"))
InvLmm = lmer(InvSimp ~ TRT + Block + (1|Block:ML), alpha %>% filter(Species == "Ip"))
SimLmm = lmer(sim ~ TRT + Block + (1|Block:ML), alpha %>% filter(Species == "Ip"))
EvenLmm = lmer(even ~ TRT + Block + (1|Block:ML), alpha %>% filter(Species == "Ip"))

anova(RichLmm)
ranova(RichLmm)

anova(InvLmm)
ranova(InvLmm)

anova(SimLmm)
ranova(SimLmm)

anova(EvenLmm)
ranova(EvenLmm)

# Competitor as fixed effect

RichLmm = lm(rich~TRT+Block*Comp,alpha %>% filter(Species == "Ip"))
InvLmm = lm(InvSimp~TRT+Block*Comp,alpha %>% filter(Species =="Ip"))
SimLmm = lm(sim~TRT+Block*Comp,alpha %>% filter(Species == "Ip"))
EvenLmm = lm(even~TRT+Block*Comp,alpha %>% filter(Species == "Ip"))

anova(RichLmm)

anova(InvLmm)

anova(SimLmm)

anova(EvenLmm)
```


## Continous data (root traits, leaf number)
Here, I correlate leaf number to OTUs with at least 5% relative abundance in at least one sample in our data set.

```{r Analyzing leaf data,warning=F,comment=F,message=F}
LeafData<-read.csv("../DataSets/SizeFitData.csv")

LeafData$Sample_ID=paste(LeafData$Position,ifelse(grepl("Ihed",LeafData$ML),"H","P"),sep="")
```


```{r Alpha, more estimates,warning=F,comment=F,message=F}

## Additional estimates of alpha diversity
 # Specify a few:
AlphaExtra = estimate_richness(physeq1, measures=c("Observed", "InvSimpson", "Shannon", "Chao1"))
```


## Preliminary test with roots

```{r Preliminary exam on root linear model}
RootData = read.csv("../DataSets/RootTraits_PCs.csv")

RootAlphaObs=merge(alpha,RootData[c("Sample_ID","PC1","PC2","PC3","PC4")])

SimpPC1=lm(sim~PC1+Species+TRT+Block,RootAlphaObs) 
summary(SimpPC1)

SimpInvPC1=lm(InvSimp~PC1+Species+TRT+Block,RootAlphaObs) 
summary(SimpInvPC1)

RichPC1=lm(rich~PC1+Species+TRT+Block,RootAlphaObs) 
summary(RichPC1)# Significant coefficient

# Adjust p-value for multiple comparisons (1 for each PC used to test)
p.adjust((summary(RichPC1)$coefficients[2,4]), method = "bonferroni", n = 4) # still significant, corrected p-value:
# 0.01158722

EvenPC1=lm(even~PC1+Species+TRT+Block,RootAlphaObs) 
summary(EvenPC1)# Significant coefficient

    # Adjust p-value for multiple comparisons (1 for each PC used to test)
    p.adjust((summary(EvenPC1)$coefficients[2,4]), method = "bonferroni", n = 4) # still significant, corrected p-value:
    # 0.02191386

SimpPC2=lm(sim~PC2+Species+TRT+Block,RootAlphaObs) 
summary(SimpPC2) 

SimpInvPC2=lm(InvSimp~PC2+Species+TRT+Block,RootAlphaObs)
summary(SimpInvPC2) 

RichPC2=lm(rich~PC2+Species+TRT+Block,RootAlphaObs) 
summary(RichPC2) # Significant coefficient

    # Adjust p-value for multiple comparisons (1 for each PC used to test)
    p.adjust((summary(RichPC2)$coefficients[2,4]), method = "bonferroni", n = 4) # still significant, corrected p-value:
    # 0.04698757


EvenPC2=lm(even~PC2+Species+TRT+Block,RootAlphaObs) 
summary(EvenPC2) # Significant coefficient

    # Adjust p-value for multiple comparisons (1 for each PC used to test)
    p.adjust((summary(EvenPC2)$coefficients[2,4]), method = "bonferroni", n = 4) # Not significant, corrected p-value:
    # 0.1492003

### PC3
SimpPC3=lm(sim~PC3+Species+TRT+Block,RootAlphaObs) 
summary(SimpPC3) 

SimpInvPC3=lm(InvSimp~PC3+Species+TRT+Block,RootAlphaObs) 
summary(SimpInvPC3) 

RichPC3=lm(rich~PC3+Species+TRT+Block,RootAlphaObs) 
summary(RichPC3)

EvenPC3=lm(even~PC3+Species+TRT+Block,RootAlphaObs) 
summary(EvenPC3)

### PC4
SimpPC4=lm(sim~PC4+Species+TRT+Block,RootAlphaObs) 
summary(SimpPC4) # Significant coefficient

    # Adjust p-value for multiple comparisons (1 for each PC used to test)
    p.adjust((summary(SimpPC4)$coefficients[2,4]), method = "bonferroni", n = 4) # still significant, corrected p-value:
    # 0.00834712

SimpInvPC4=lm(InvSimp~PC4+Species+TRT+Block,RootAlphaObs) 
summary(SimpInvPC4)  # Significant coefficient
  
  # Adjust p-value for multiple comparisons (1 for each PC used to test)
      p.adjust((summary(SimpInvPC4)$coefficients[2,4]), method = "bonferroni", n = 4) # marginally significant, corrected p-value:
      # 0.07073862

RichPC4=lm(rich~PC4+Species+TRT+Block,RootAlphaObs) 
summary(RichPC4)

EvenPC4=lm(even~PC4+Species+TRT+Block,RootAlphaObs) 
summary(EvenPC4)
```

## Within species (root traits and alphadiv)


```{r Preliminary exam on roots within species}

library(multcomp)

#summary(glht(SimpInvPC1, mcp(rank="Tukey")))

#################################################################
##################### SUBSET for I.purpurea #####################
#################################################################


RootAlphaPurp=droplevels(RootAlphaObs%>%filter(Species=="Ip"))
RootAlphaPurp$Comp=sub(".*\\-","",RootAlphaPurp$Combos)

SimpPC1=lm(sim~PC1+Comp+TRT+Block,RootAlphaPurp) 
summary(SimpPC1)
anova(SimpPC1) # Significant Competitor effect


SimpInvPC1=lm(InvSimp~PC1+Comp+TRT+Block,RootAlphaPurp) 
anova(SimpInvPC1) # Significant PC1 effect only, no Comp or Comp by PC1 effect


RichPC1=lm(rich~PC1+Comp+TRT+Block,RootAlphaPurp) 
anova(RichPC1) # Marginal Competitor effect


anova(lm(rich~PC1+Comp+TRT+Block,RootAlphaPurp))

EvenPC1=lm(even~PC1+Comp+TRT+Block,RootAlphaPurp) 
anova(EvenPC1)



SimpPC2=lm(sim~PC2+Comp+TRT+Block,RootAlphaPurp) 
anova(SimpPC2) # NS

# ANCOVA
anova(lm(sim~PC2*TRT+Block,RootAlphaPurp)) # NS

SimpInvPC2=lm(InvSimp~PC2+Comp+TRT+Block,RootAlphaPurp) 
anova(SimpInvPC2) # NS


RichPC2=lm(rich~PC2+Comp+TRT+Block,RootAlphaPurp) 
anova(RichPC2) # Significant coefficient SUPER small effect size


EvenPC2=lm(even~PC2+Comp+TRT+Block,RootAlphaPurp) 
anova(EvenPC2) # Significant mean Evenness effect--should Check this out it is suspect


### PC3
SimpPC3=lm(sim~PC2+Comp+TRT+Block,RootAlphaPurp) 
anova(SimpPC3) # NS


SimpInvPC3=lm(InvSimp~PC3+Comp+TRT+Block,RootAlphaPurp) 
anova(SimpInvPC3) # NS

RichPC3=lm(rich~PC3+Comp+TRT+Block,RootAlphaPurp) 
anova(RichPC3)

EvenPC3=lm(even~PC3+Comp+TRT+Block,RootAlphaPurp) 
summary(EvenPC3)
anova(EvenPC3)


### PC4
SimpPC4=lm(sim~PC4+Comp+TRT+Block,RootAlphaPurp%>%filter(sim>0.94)) 
summary(SimpPC4) # Significant
anova(SimpPC4)


SimpInvPC4=lm(InvSimp~PC4+Comp+TRT+Block,RootAlphaPurp) 
summary(SimpInvPC4)  # Marginal InvSimp significant
anova(SimpInvPC4)

RichPC4=lm(rich~PC4+Comp+TRT+Block,RootAlphaPurp) 
summary(RichPC4) # NS
anova(RichPC4)

EvenPC4=lm(even~PC4+Comp+TRT+Block,RootAlphaPurp) 
summary(EvenPC4)
anova(EvenPC4)


```

## Plotting significant linear associations
```{r n}

# Richness and root architecture
P2.rich=ggplot()+
  geom_point(data=RootAlphaPurp,aes(PC2,rich),alpha=0.5,size=5)+
  geom_smooth(data=RootAlphaPurp,method="lm",aes(PC2,rich),fullrange=TRUE)+
  theme_classic()+
  ylab("Richness")+
  xlab("")+
  Tx2
#+ 
  #annotate("text", x = -1, y = 500, label = "paste(italic(R) ^ 2, \" = 0.15\")", parse = TRUE,hjust=0,size=5)+ 
 # annotate("text", x = -1, y = 490, label = "paste(italic(B), \" = -4.65 +/-1.81\")", parse = TRUE,hjust=0,size=5)+
 # annotate("text", x = -1, y = 480, label = "paste(italic(P), \" = 0.01\")", parse = TRUE,hjust=0,size=5)

# Evenness and root architecture
P2.even=ggplot()+
  geom_point(data=RootAlphaPurp,aes(PC2,even),alpha=0.5,size=5)+
  geom_smooth(data=RootAlphaPurp,method="lm",aes(PC2,even),fullrange=TRUE)+
  theme_classic()+
  ylab("Evenness")+
  xlab("")+
  Tx2#+ 
  #annotate("text", x = -4.5, y = .0114, label = "paste(italic(R) ^ 2, \" = 0.12\")", parse = TRUE,hjust=0,size=5)+ 
#  annotate("text", x = -4.5, y = .0112, label = "paste(italic(B), \" = -7.29E-5 +/-3.28E-5\")", parse = TRUE,hjust=0,size=5)+
 # annotate("text", x = -4.5, y =.011, label = "paste(italic(P), \" = 0.04\")", parse = TRUE,hjust=0,size=5)


# Root morphology on species diversity Simpson metric

P4.Sim<-ggplot()+
  geom_point(data=RootAlphaPurp,aes(PC4,sim),alpha=0.5,size=5)+
  geom_smooth(data=RootAlphaPurp,method="lm",aes(PC4,sim),fullrange=TRUE)+
  theme_classic()+
  ylab("Simpson")+
  xlab("")+
  Tx2#+
  #annotate("text", x = -5, y = .988, label = "paste(italic(R) ^ 2, \" = 0.18\")", parse = TRUE,hjust=0,size=5)+ 
  #annotate("text", x = -5, y = .985, label = "paste(italic(B), \" = 2.22E-4 +/-6.99E-4\")", parse = TRUE,hjust=0,size=5)+
 # annotate("text", x =-5, y = .982, label = "paste(italic(P), \" < 0.01\")", parse = TRUE,hjust=0,size=5)


P4.simIn=ggplot()+
  geom_point(data=RootAlphaPurp,aes(PC4,InvSimp),alpha=0.5,size=5)+
  geom_smooth(data=RootAlphaPurp,method="lm",aes(PC4,InvSimp),fullrange=TRUE)+
  theme_classic()+
  ylab("Inverse Simpson")+
  xlab("")+
  Tx2#+
 # annotate("text", x = -5, y = 60, label = "paste(italic(R) ^ 2, \" = 0.12\")", parse = TRUE,hjust=0,size=5)+ 
 # annotate("text", x = -5, y = 57, label = "paste(italic(B), \" = 2.08 +/-1.05\")", parse = TRUE,hjust=0,size=5)+
 # annotate("text", x =-5, y = 54, label = "paste(italic(P), \" = 0.02\")", parse = TRUE,hjust=0,size=5)


```


## Linear mixed models

```{r}
library(lmerTest)

### Simpson
SimpLMM=lmer(sim~TRT+Block+(1|ML),alpha%>%filter(Species=="Ip")) 
anova(SimpLMM)
ranova(SimpLMM)

### Inverse Simpson
SimpInvLMM=lmer(InvSimp~TRT+Block+(1|ML),alpha%>%filter(Species=="Ip")) 
anova(SimpInvLMM)
ranova(SimpInvLMM)

### Inverse Simpson
RichLMM=lmer(rich~TRT+Block+(1|ML),alpha%>%filter(Species=="Ip")) 
anova(RichLMM)
ranova(RichLMM)

# Bray curtis metric now


Ax1LMM=lmer(Axis.1~Species+TRT+Block+(1|ML),physeq.pcoa.df)
anova(Ax1LMM)
ranova(Ax1LMM)


Ax2LMM=lmer(Axis.2~Species+TRT+Block+(1|ML),physeq.pcoa.df)
anova(Ax2LMM) # Significant species diffeences in axis 2
ranova(Ax2LMM)


Ax3LMM=lmer(Axis.3~Species+TRT+Block+(1|ML),physeq.pcoa.df)
anova(Ax3LMM) 
ranova(Ax3LMM)

Ax4LMM=lmer(Axis.4~Species+TRT+Block+(1|ML),physeq.pcoa.df)
anova(Ax4LMM) # Marginal species differences
ranova(Ax4LMM)
```


```{r Read in fitness data}
Fitness=read.csv("../DataSets/FitPA4.csv")

library(dplyr)
# Calculate relative fitness
  # First calculate mean seed number by species and treatment---note* we only have seed output of I. purpurea
MeanSeedNumber=aggregate(SeedNumber~Trt+Species,Fitness,mean)

colnames(MeanSeedNumber)=c("Trt","Species","MeanSeedNumber")

Ipurp.Fit=Fitness%>%
  filter(Species=="Ip")

Ipurp.Alpha=alpha%>%
  filter(Species=="Ip")

FitnessPurp <- merge(Ipurp.Fit,MeanSeedNumber,by=c("Trt","Species"))
FitnessPurp$RelativeFit=FitnessPurp$SeedNumber/FitnessPurp$MeanSeedNumber
FitnessPurp$Block=as.factor(FitnessPurp$Block)

FitnessPurp2=merge(FitnessPurp,LeafData)
str(FitnessPurp2)
FitnessPurp2$Leaf.Number=as.numeric(as.character(FitnessPurp2$Leaf.Number))

SN1<-lmer(SeedNumber~Trt+Block+Leaf.Number+Block:Trt+(1|ML),FitnessPurp2)
anova(SN1) # Treatment is significant effect on plant seed number; Significant Treatment by Block effect

ggplot(FitnessPurp,aes(Trt,RelativeFit,fill=Block))+
  geom_boxplot()+
  scale_fill_brewer("Paired")+
  theme_classic()+
  ylab("Relative Fitness")+
  ggtitle("Relative Fitness by Treatment and Block")

# Remove size effects from fitness
StdFitness=FitnessPurp2[c("Trt","Species","Block","ML","RelativeFit","Leaf.Number")] # Subset fitness data for variables of interest

# Run one-way ANOVA to remove size effect--i.e., keep residuals
StdFitness$RelativeFitness=residuals(lm(RelativeFit~Leaf.Number,FitnessPurp2)) 

# Compare residuals and non-standard values of fitness
dev.off()
plot(StdFitness$RelativeFit,StdFitness$RelativeFitness)

# Average fitness by block, maternal line and treatment--we use NON standardized fitness (ie size effects not removed)
FitAveraged <- aggregate(RelativeFit~Block+Trt+ML,FitnessPurp,mean)
colnames(FitAveraged)<-c("Block","Trt","ML","RelativeFitness")
FitAveraged$TRT=FitAveraged$Trt
dim(FitAveraged)
head(FitAveraged)
```


```{r phenotypic selection}

####  ####  ####  ####  ####  ####   ####
#  Examine selection on microbiome first
####  ####  ####  ####  ####  ####   ####
FitnessPurp$TRT=FitnessPurp$Trt
FitnessPurp$Combos=as.character(FitnessPurp$Combos)
FitnessPurp[which(FitnessPurp$Trt=="Alone"),]$Combos="none"
FitnessPurp$Combos=as.factor(FitnessPurp$Combos)

BrayFit=merge(physeq.pcoa.df,FitAveraged)

FitAlpha <- merge(FitAveraged,alpha)
dim(FitAlpha)

ggplot(FitAlpha,aes(TRT,RelativeFitness))+
  geom_boxplot()+
  scale_fill_brewer("Paired")+
  theme_classic()+
  ylab("Relative Fitness")+
  ggtitle("Relative Fitness by Treatment and Block")+
  facet_grid(~Block)


# Combine with root data
library(dplyr)

RootAveraged=aggregate(list(RootData[c("PC1","PC2","PC3","PC4")]),by=list(RootData$Trt,RootData$ML),FUN=mean) 

colnames(RootAveraged)=c("Trt","ML","PC1","PC2","PC3","PC4")
head(RootAveraged)

RootFitAlpha <- merge(FitAlpha,RootAveraged)
head(RootFitAlpha)

#RootFitLeafAlpha=merge(RootFitAlpha,LeafData[c("Sample_ID","Leaf.Number")],by="Sample_ID")

#RootFitLeafAlpha$Leaf.Number=as.numeric(as.character(RootFitLeafAlpha$Leaf.Number))


# CombinE root,fitness/bray estimates
RootFitBray=merge(BrayFit,RootAveraged)


# Plot boxplots of averaged root traits
PC2_Box=ggplot(RootAveraged,aes(x="",y=PC2))+
  geom_boxplot()+
  #geom_jitter()+
  xlab("Root Architecture")+
  theme_classic()+
  Tx+ 
  coord_flip()

# Plot boxplots of averaged root traits
PC1_Box=ggplot(RootAveraged,aes(x="",y=PC1))+
  geom_boxplot()+
  #geom_jitter()+
  xlab("Root Topology")+
  theme_classic()+
  Tx+ 
  coord_flip()

# Plot boxplots of averaged root traits
PC4_Box=ggplot(RootAveraged,aes(x="",y=PC4))+
  geom_boxplot()+
  #geom_jitter()+
  xlab("Root Morphology")+
  theme_classic()+
  Tx+ 
  coord_flip()

#ggarrange(P2.rich,P2.even,P4.Sim,P4.simIn,nrow=2,ncol=2)
AB=cowplot::plot_grid(P2.rich,P2.even,align = "hv", ncol = 2,labels =c("A","B"), label_size =22 ,hjust=-11,vjust=1,label_x = -0.16)

ab=cowplot::plot_grid(P4.Sim,P4.simIn,align = "hv", ncol = 2,labels =c("A","B"), label_size =22 ,hjust=-11,vjust=1,label_x = -0.17)


P4.simIn+
  xlab("Root morphology (PC4)")
# Common x title

x.grob <- textGrob("Root morphology (PC4)", 
                   gp=gpar(col="black", fontsize=25), rot=0)

gridExtra::grid.arrange(gridExtra::arrangeGrob(ab,bottom=x.grob,padding = unit(0.5,units = 'in'),nrow=1))

# Common x title
x.grob1 <- textGrob("Root architecture (PC2)", 
                   gp=gpar(col="black", fontsize=25), rot=0)

gridExtra::grid.arrange(gridExtra::arrangeGrob(AB,bottom=x.grob1,padding = unit(0.5,units = 'in'),nrow=1))

```


## Selection on microbe variables

```{r}
# Quadratic microbe term
FitAlpha$even2=(FitAlpha$even*FitAlpha$even)
FitAlpha$rich2=(FitAlpha$rich*FitAlpha$rich)
FitAlpha$InSim2=(FitAlpha$InvSimp*FitAlpha$InvSimp)
FitAlpha$sim2=(FitAlpha$sim*FitAlpha$sim)

```

### Selection on richness
```{r cor selection on richness}
summary(lm(RelativeFitness~Block+rich+rich2,FitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+rich+rich2,FitAlpha%>%filter(TRT!="Alone")))

richness_sel=lm(RelativeFitness~TRT*Block+rich,FitAlpha)
summary(richness_sel) # No evidence of indirect selection
anova(richness_sel)


ggplot(FitAlpha)+
  geom_point(aes(RelativeFitness,rich),alpha=0.5,size=5)+
  geom_smooth(method="lm",aes(RelativeFitness,rich),fullrange=TRUE)+
  theme_classic()+
  ylab("Relative fitness")+
  xlab("Richness")+
  Tx2

```


### Selection on Inverse Simpson
```{r cor selection on Inverse Simpson}
summary(lm(RelativeFitness~Block+InvSimp+InSim2,FitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+InvSimp+InSim2,FitAlpha%>%filter(TRT!="Alone")))

Invn_sel=lm(RelativeFitness~TRT*Block+InvSimp+Block,FitAlpha)
summary(Invn_sel) # No evidence of indirect selection

ggplot(FitAlpha)+
  geom_point(aes(RelativeFitness,InvSimp),alpha=0.5,size=5)+
  geom_smooth(method="lm",aes(RelativeFitness,InvSimp),fullrange=TRUE)+
  theme_classic()+
  ylab("Relative fitness")+
  xlab("Inverse Simpson")+
  Tx2
```

## Selection on Simpson
```{r cor selection on Simpson }
summary(lm(RelativeFitness~Block+sim+sim2,FitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+sim+sim2,FitAlpha%>%filter(TRT!="Alone")))

sim_sel=lm(RelativeFitness~TRT*Block+sim,FitAlpha)
summary(sim_sel) # NS

ggplot(FitAlpha)+
  geom_point(aes(RelativeFitness,sim),alpha=0.5,size=5)+
  geom_smooth(method="lm",aes(RelativeFitness,sim),fullrange=TRUE)+
  theme_classic()+
  ylab("Relative fitness")+
  xlab("Simpson")+
  Tx2
```

## Selection on Evenness
```{r cor selection on Evenness}

summary(lm(RelativeFitness~Block+even+even2 ,FitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+even+even2,FitAlpha%>%filter(TRT!="Alone"))) # Significant quadratic term for evenness on fitnes (super low slope tho)



even_sel=lm(RelativeFitness~Trt*Block+TRT*even2+even,FitAlpha) # if we include block by treatment interaction there is marginal significance for a treatment by evenness squared effect
even_sel2=lm(RelativeFitness~Block+TRT*even2+even,FitAlpha) # if we include block by treatment interaction there is marginal significance for a treatment by evenness squared effect

anova(even_sel,even_sel2) # Based on RSS--second model/simpler one is a bit better
anova(even_sel) # NS

ggplot(FitAlpha)+
  geom_point(aes(RelativeFitness,even),alpha=0.5,size=5)+
  stat_smooth(method="lm",formula = y ~poly(x,2),aes(RelativeFitness,even))+
  theme_classic()+
  ylab("Relative fitness")+
  xlab("Evenness")+
  Tx2
```

```{r Tests of Phenotypic Selection}
RootFitAlpha$Comp<-sub(".*\\-","",RootFitAlpha$Combos)

# PC1 and richness
summary(lm(RelativeFitness~Block+PC1*rich,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+PC1*rich+Comp,RootFitAlpha%>%filter(TRT!="Alone")))
anova(lm(RelativeFitness~Block+PC1*rich+Comp,RootFitAlpha%>%filter(TRT!="Alone")))
anova(lm(RelativeFitness~Block*Trt+PC1*rich+Trt:Comp,RootFitAlpha))

richPC1=lm(RelativeFitness~PC1*rich*Trt+Block,RootFitAlpha)
summary(richPC1)

# PC2 and richness
summary(lm(RelativeFitness~PC2*rich*Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC2*rich*Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC2*rich*TRT*Block,RootFitAlpha)) # Interesting results!  effect
# Phenotypic selection on root architecture according to block and treatment--significant interaction between PC2, richness, treatment and block

richPC2=lm(RelativeFitness~PC2*rich*TRT+Block,RootFitAlpha)
summary(richPC2)

# PC3 and richness
summary(lm(RelativeFitness~Block*PC3*rich,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block*PC3*rich,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC3*rich*TRT*Block,RootFitAlpha)) # Significant richness effect

# Significant Block effects
# PC3:rich:Block Significant
# TRT:Block significant
# PC3:rich:TRT significant

# PC4 and richness
summary(lm(RelativeFitness~Block*PC4*rich,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block*PC4*rich,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC4*rich*TRT*Block,RootFitAlpha))

# Significant PC4:Block
# PC4:Block
# Trt:Block
# rich:TRT:Block
# PC4:rich:TRT:Block
# Block

# PC3 and simpson
summary(lm(RelativeFitness~Block+PC3*sim,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+PC3*sim,RootFitAlpha%>%filter(TRT!="Alone")))
anova(lm(RelativeFitness~PC1*sim*TRT*Block,RootFitAlpha))

# PC1:TRT:Block Significant
# PC1:Block
# Block


# PC2 and simpson
summary(lm(RelativeFitness~Block+PC2*sim,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+PC2*sim,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC2*sim*TRT*Block,RootFitAlpha))

# Block sig
# PC2:TRT:Block significant
# PC2:sim:TRT:Block
# Marginal treatment effect 


# PC3 and simpson
summary(lm(RelativeFitness~Block+PC3*sim,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+PC3*sim,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC3*sim*TRT*Block,RootFitAlpha))
# Marginal simpson
# significant Block
# TRT:Block significant
# PC3:sim:TRT significant
# PC3:TRT:Block 


# PC4 and simpson
summary(lm(RelativeFitness~Block+PC4*sim,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~Block+PC4*sim,RootFitAlpha%>%filter(TRT!="Alone")))


anova(lm(RelativeFitness~PC4*sim*TRT*Block,RootFitAlpha))

# Significant PC4
# Significant Block
# Significant PC4:Block
# Significant sim:TRT:Block


# Inv Simp and PC1
summary(lm(RelativeFitness~PC1*InvSimp+Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC1*InvSimp+Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC1*InvSimp*TRT+Block,RootFitAlpha))
# Significant InvSimp
# Significant Block

# Inv Simp and PC2
summary(lm(RelativeFitness~PC2*InvSimp+Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC2*InvSimp+Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC2*InvSimp*TRT*Block,RootFitAlpha))
# Significant Inverse simpson 
# Significant Block
# Significant PC2:TRT:Block 
# Significant PC2:InvSimp:TRT:Block 
# Marginal InvSimp:TRT:Block

# PC3 and Inv Sipm
anova(lm(RelativeFitness~PC3*InvSimp*TRT+Block,RootFitAlpha))

# summary(InvSimpPC3) #Significant

# Within treatment regressions
#InvSimpPC3Alone = lm(RelativeFitness~PC3*InvSimp*Comp+Block,RootFitAlpha%>%filter(TRT=="Alone"))

#summary(InvSimpPC3Alone)

InvSimpPC3Comp = lm(RelativeFitness~PC3*InvSimp+Block,RootFitAlpha %>%filter(TRT!="Alone"))

summary(InvSimpPC3Comp)

# Inv simp and PC4
summary(lm(RelativeFitness~PC4*InvSimp+Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC4*InvSimp+Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC4*InvSimp*TRT*Block,RootFitAlpha))
# Significant PC4
# Significant InvSimp
# Significant Block
# Significant PC4:Block
# Significant InvSimp:TRT:Block





# Even and PC1
summary(lm(RelativeFitness~PC1*even+Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC1*even+Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC1*even*TRT*Block,RootFitAlpha))
# Marginal PC1:even:Block
# Significant PC1:TRT:Block 
# Significant PC1:Block 
# Significant even 
# Significant Block

# Even and PC2
summary(lm(RelativeFitness~PC2*even+Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC2*even+Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC2*even*TRT*Block,RootFitAlpha))
# Significant even
# Significant Treatment
# Significant Block
# Significant PC2:TRT:Block
# Significant PC2:even:TRT:Block

# Even and PC3
summary(lm(RelativeFitness~PC3*even+Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC3*even+Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC3*even*TRT*Block,RootFitAlpha))
# Significant even
# Significant Block
# Significant TRT:Block
# Significant PC3:even:Block
# Marginal PC3:TRT:Block  

# Even and PC4
summary(lm(RelativeFitness~PC4*even+Block,RootFitAlpha%>%filter(TRT=="Alone")))
summary(lm(RelativeFitness~PC4*even+Block,RootFitAlpha%>%filter(TRT!="Alone")))

anova(lm(RelativeFitness~PC4*even*TRT*Block,RootFitAlpha))
# Significant even
# Significant PC4
# Marginal TRT
# Significant Block
# PC4:Block
# Marginal Block:TRT
# Significant PC4:even:TRT:Block
```


# MANTEL test to examine how microbiome disimilarity is correlated root phenotype distance
```{r,include=F,render=F}
# Here I use observed values of root traits 

# Load matrix Bray distances of OTU abundance
OTU_table=t(otu_table(physeq1)) # Write out OTU table


# Pull out Root traits of interest and save these to the Sampled data frame
SampledRoots=merge(RootAlphaObs,sampledf)


OTU_table=OTU_table[row.names(OTU_table)%in%SampledRoots$Sample_ID,]


    OTU=OTU_table
    Roots=SampledRoots

  # Calculate root architecture distances with euclidean distance 

    PC2=Roots$PC2 # isolate PC2, i.e. root architecture
    PC2.dist=dist(PC2)

  # Calculate Bray distance matrix for OTU table

    Bray=vegdist(OTU,method="bray")


    
# OTU Bray vs Root architecture

    OTU_pc2 = mantel(Bray, PC2.dist, method = "spearman", permutations = 9999, na.rm = TRUE)

    # Marginally significant--very low r value
    # Mantel statistic r: 0.03795 
    # Significance: 0.1166 
    
# Reapat for root topology, size and morphology

  PC1=Roots$PC1 # isolate 
  PC1.dist=dist(PC1)
  PC3=Roots$PC3 # isolate 
  PC3.dist=dist(PC3)
  PC4=Roots$PC1 # isolate 
  PC4.dist=dist(PC4)

# Examine overall distances in root system; i.e. use all PCs in distance calculation

    PCall=Roots[grep("PC",names(Roots))]
    PC.dist=dist(PCall)

    OTU_pc = mantel(Bray, PC.dist, method = "spearman", permutations = 9999, na.rm = TRUE)

###             examine mantel test using observed values of roots ###

OTU_pc1 = mantel(Bray,PC1.dist, method = "spearman", permutations = 9999, na.rm = TRUE)
OTU_pc2 = mantel(Bray,PC2.dist, method = "spearman", permutations = 9999, na.rm = TRUE)
OTU_pc3 = mantel(Bray,PC3.dist, method = "spearman", permutations = 9999, na.rm = TRUE)
OTU_pc4 = mantel(Bray,PC4.dist, method = "spearman", permutations = 9999, na.rm = TRUE)

  # Repeat but standardize for Block
  
  Standardize=function(x,data){
  StandData=residuals(lm(x~Block,data))
  return(StandData)
  }
  
  PC1_resid=Standardize(x=PC1,data=Roots)
  PC2_resid=Standardize(x=PC2,data=Roots)
  PC3_resid=Standardize(x=PC1,data=Roots)
  PC4_resid=Standardize(x=PC1,data=Roots)
  
  
    PC1.distR=dist(PC1_resid)
    PC2.distR=dist(PC2_resid)
    PC3.distR=dist(PC3_resid)
    PC4.distR=dist(PC4_resid)
  
  
  
  OTU_pc1_st = mantel(Bray,PC1.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  OTU_pc2_st = mantel(Bray,PC2.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  OTU_pc3_st = mantel(Bray,PC3.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  OTU_pc4_st = mantel(Bray,PC4.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  
  # RE RUN WITHIN I. PURP ONLY
  
# Pull out Root traits of interest and save these to the Sampled data frame
SampledRoots=merge(RootAlphaObs,sampledf)
SampledRootsIp=subset(SampledRoots,Species=="Ip")

OTU_table=OTU_table[row.names(OTU_table)%in%SampledRootsIp$Sample_ID,]


    OTU=OTU_table
    Roots=SampledRootsIp

  # Calculate root architecture distances with euclidean distance 

    PC2=Roots$PC2 # isolate PC2, i.e. root architecture
    PC2.dist=dist(PC2)

  # Calculate Bray distance matrix for OTU table

    Bray=vegdist(OTU,method="bray")


    
# OTU Bray vs Root architecture

    OTU_pc2 = mantel(Bray, PC2.dist, method = "spearman", permutations = 9999, na.rm = TRUE)

    # Marginally significant--very low r value
    # Mantel statistic r: 0.06836  
    # Significance: 0.07
    
# Reapat for root topology, size and morphology

  PC1=Roots$PC1 # isolate 
  PC1.dist=dist(PC1)
  PC3=Roots$PC3 # isolate 
  PC3.dist=dist(PC3)
  PC4=Roots$PC1 # isolate 
  PC4.dist=dist(PC4)

# Examine overall distances in root system; i.e. use all PCs in distance calculation

    PCall=Roots[grep("PC",names(Roots))]
    PC.dist=dist(PCall)

    OTU_pc = mantel(Bray, PC.dist, method = "spearman", permutations = 9999, na.rm = TRUE)

###             examine mantel test using observed values of roots ###

OTU_pc1 = mantel(Bray,PC1.dist, method = "spearman", permutations = 9999, na.rm = TRUE)
# OTU_pc2 = mantel(Bray,PC2.dist, method = "spearman", permutations = 9999, na.rm = TRUE)
OTU_pc3 = mantel(Bray,PC3.dist, method = "spearman", permutations = 9999, na.rm = TRUE)
OTU_pc4 = mantel(Bray,PC4.dist, method = "spearman", permutations = 9999, na.rm = TRUE)

# Repeat but standardize for Block
  
  Standardize=function(x,data){
  StandData=residuals(lm(x~Block,data))
  return(StandData)
  }
  
  PC1_resid=Standardize(x=PC1,data=Roots)
  PC2_resid=Standardize(x=PC2,data=Roots)
  PC3_resid=Standardize(x=PC1,data=Roots)
  PC4_resid=Standardize(x=PC1,data=Roots)
  
  
    PC1.distR=dist(PC1_resid)
    PC2.distR=dist(PC2_resid)
    PC3.distR=dist(PC3_resid)
    PC4.distR=dist(PC4_resid)
  
  
  
  OTU_pc1_st = mantel(Bray,PC1.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  OTU_pc2_st = mantel(Bray,PC2.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  OTU_pc3_st = mantel(Bray,PC3.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  OTU_pc4_st = mantel(Bray,PC4.distR, method = "spearman", permutations = 9999, na.rm = TRUE)
  
```




### MANTEL between Bray Curtis and Relative Fitnesses
```{r}
#  Here we use the Family mean values of root traits---this would indicate evidence for 'phenotypic selection' 

  # Load matrix Bray distances of OTU abundance
  OTU_table=t(otu_table(physeq1)) # Write out OTU table
  

  # Pull out Root traits of interest and save these to the Sampled data frame
  SampledFit=merge(FitAveraged,sampledf,by=c("TRT","ML","Block"))

  # Subset per treatment
  
SampledFitComp=SampledFit%>%filter(TRT=="Inter")
OTU_table_Comp=OTU_table[row.names(OTU_table)%in%SampledFitComp$Sample_ID,]

SampledFitAlone=SampledFit%>%filter(TRT!="Inter")
OTU_table_Alone=OTU_table[row.names(OTU_table)%in%SampledFitAlone$Sample_ID,]

    OTU_Comp=OTU_table_Comp
    Fit_Comp=SampledFitComp
  
    OTU_Alone=OTU_table_Alone
    Fit_Alone=SampledFitAlone
    # Calculate fitness distances w euclidean distance 

    Fit.distComp=dist(Fit_Comp$RelativeFitness)
    Fit.distAlone=dist(Fit_Alone$RelativeFitness)

    # Calculate Bray distance matrix for OTU table
    BrayComp=vegdist(OTU_Comp,method="bray")
    BrayAlone=vegdist(OTU_Alone,method="bray")


    # OTU Bray vs Fitness per treatment

    OTU_Fit_Comp = mantel(BrayComp, Fit.distComp, method = "spearman", permutations = 9999, na.rm = TRUE)
    OTU_Fit_Alone = mantel(BrayAlone, Fit.distAlone, method = "spearman", permutations = 9999, na.rm = TRUE)
    
    
# Run partial mantel test and control for plant size--i.e., plant size distance matrix
    
    # First  average combined fitness and leaf data 
    Fit2Average=FitnessPurp2%>%
      filter(Species=="Ip")%>%
      group_by(ML,Block,Trt)%>%
      summarise("RelativeFitAv"=mean(RelativeFit,rm.na=T),"LeafAv"=mean(Leaf.Number,rm.na=T))
    Fit2Average$TRT=Fit2Average$Trt
  
    # Pull out Root traits of interest and save these to the Sampled data frame
  SampledFit=merge(Fit2Average,sampledf,by=c("TRT","ML","Block"))

# Subset per treatment
  
  SampledFitComp=SampledFit%>%filter(TRT=="Inter")
  OTU_table_Comp=OTU_table[row.names(OTU_table)%in%SampledFitComp$Sample_ID,]
  
  SampledFitAlone=SampledFit%>%filter(TRT!="Inter")
  OTU_table_Alone=OTU_table[row.names(OTU_table)%in%SampledFitAlone$Sample_ID,]

    OTU_Comp=OTU_table_Comp
    Fit_Comp=SampledFitComp
  
    Fit_Alone=SampledFitAlone
    
    OTU_Alone=OTU_table_Alone

     # Remove NAS
    Remove=which(Fit_Alone$LeafAv%in%NA)
    
    # Calculate fitness distances w euclidean distance 

    Fit.distComp=dist(Fit_Comp$RelativeFitAv)
    Fit.distAlone=dist(Fit_Alone[-Remove,"RelativeFitAv"])
    
    # Calculate size distances w euclidean distance 

    Leaf.distComp=dist(Fit_Comp$LeafAv)
    Leaf.distAlone=dist(Fit_Alone[-Remove,"LeafAv"])


    # Calculate Bray distance matrix for OTU table
    BrayComp=vegdist(OTU_Comp,method="bray")
    BrayAlone=vegdist(OTU_Alone[-Remove,],method="bray")


    # OTU Bray vs Fitness per treatment

    OTU_Fit_Comp_Par = mantel.partial(BrayComp, Fit.distComp, Leaf.distComp, method = "spearman", permutations = 9999, na.rm = TRUE)
    
    OTU_Fit_Alone_Par = mantel.partial(BrayAlone, Fit.distAlone,Leaf.distAlone, method = "spearman", permutations = 9999, na.rm = TRUE)
```



# Weighted network analysis
```{r,WGNC compare between species}

# Filter for counts that are greater than 10 
filter <- phyloseq::genefilter_sample(physeq1, filterfun_sample(function(x) x >= 10))#,A = 0.2*nsamples(physeq1))

phy_filtered <- prune_taxa(filter, physeq1)
OTU_count=t(otu_table(phy_filtered))


# Load the WGCNA package
library(WGCNA)


## Subset data by species and treatment
Comp=SAMPLE[which(SAMPLE$TRT%in%"Inter"),]
CompPurp=Comp[which(Comp$Species%in%"Ip"),]
CompHed=Comp[which(Comp$Species%in%"Ihed"),]
AloneIp=SAMPLE[which(SAMPLE$TRT%in%"Alone"),]

table(CompPurp$Species)
table(CompHed$Species)
table(AloneIp$Species)


OTU_CompPurp=OTU_count[which(row.names(OTU_count)%in%CompPurp$Sample_ID),]
OTU_CompHed=OTU_count[which(row.names(OTU_count)%in%CompHed$Sample_ID),]
OTU_AloneIp=OTU_count[which(row.names(OTU_count)%in%AloneIp$Sample_ID),]

# Check for 'good' OTU's
testPurp=goodSamplesGenes(OTU_CompPurp)
testHed=goodSamplesGenes(OTU_CompHed)
testPurp.alone=goodSamplesGenes(OTU_AloneIp)

A=which(testPurp$goodGenes==FALSE) # One 'bad' OTU
B=which(testHed$goodGenes==FALSE) # Three 'bad' OTU's
C=which(testPurp.alone$goodGenes==FALSE) # Nine'bad' OTU's

# Remove the 'bad' OTUs equally in each subset data
OTU_CompPurp=OTU_CompPurp[,-c(A,B,C)]
OTU_CompHed=OTU_CompHed[,-c(A,B,C)]
OTU_AlonePurp=OTU_AloneIp[,-c(A,B,C)]

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(OTU_CompPurp, powerVector = powers, verbose = 5)

# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")

# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")

# Lets pick 8 for the soft threwshold power 


# Repeat for I. hederacea

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(OTU_CompHed, powerVector = powers, verbose = 5)

# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")

# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")

# stick with 8 which is just at threshold value

# Repeat for I. purpurea alone

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(OTU_AlonePurp, powerVector = powers, verbose = 5)

# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")

# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")

# 8 is also just above the threshold!
```


```{r, next step--run adjacency matrices for ea species}
# Work with I. purpurea and I. hederacea within competition only
softPower = 8;
adjacency_P = adjacency(OTU_CompPurp, power = softPower); # I. purpurea
adjacency_H = adjacency(OTU_CompHed, power = softPower); # I. hederacea

# Turn adjacency into topological overlap
TOM_P = TOMsimilarity(adjacency_P);
dissTOM_P = 1-TOM_P

TOM_H = TOMsimilarity(adjacency_H);
dissTOM_H = 1-TOM_H

# Hierarchical clustering:
TreeP = hclust(as.dist(dissTOM_P), method = "average");
TreeH = hclust(as.dist(dissTOM_H), method = "average");

colorh1=as.character(cutreeStaticColor(TreeP,cutHeight=.97,minSize=30))
table(colorh1)

colorh2=as.character(cutreeStaticColor(TreeH,cutHeight=.97,minSize=30))
table(colorh2)


dynamicColors_P = labels2colors(colorh1)
dynamicColors_H = labels2colors(colorh2)

# Calculate eigengenes
MEList_P = moduleEigengenes(OTU_CompPurp, colors = dynamicColors_P)
MEs_P = MEList_P$eigengenes
MEList_H = moduleEigengenes(OTU_CompHed, colors = dynamicColors_H)
MEs_H = MEList_H$eigengenes

# Calculate dissimilarity of module eigengenes
MEDiss_P = 1-cor(MEs_P);
MEDiss_H = 1-cor(MEs_H);

# Set figure layout
layout(matrix(c(1:4), 2,2), heights = c(0.8, 0.2));

# Plot the dendrograms and module colors !!! 

plotDendroAndColors(TreeP,cbind(colorh1, colorh2),c("I. purpurea", "I. hederacea"),main="I. purpurea", dendroLabels=FALSE,setLayout = FALSE, abHeight = 0.97,marAll = c(0.3, 4.5, 2, 0.2))

plotDendroAndColors(TreeH,cbind(colorh1, colorh2),c("I. purpurea", "I. hederacea"),main="I. hederacea", dendroLabels=FALSE,setLayout = FALSE, abHeight = 0.97)

# If plotting into a file, close it.dev.off();

# Save the results

colorhALL=as.character(colorh1)
colorh2ALL=as.character(colorh2)

# Give the module colors more descriptive names
colorPurp = colorhALL;colorHed = colorh2ALL;


# Save module colors (assignments) and the indicator
save(colorPurp, colorHed, file = "PurpHed-SpeciesAnalysis-colorPurp-colorHed.RData")

```


## Preservation analysis (Species)

```{r Preservation analysis between species}

OTU_CompPurp=as(OTU_CompPurp, "matrix")
OTU_CompHed=as(OTU_CompHed, "matrix")

# Object that will contain the expression data

SpData=list(Ipurp = list(data = OTU_CompPurp), Ihed = list(data = OTU_CompHed));

# Set labels
setLabels=c("Ip","Ihed")
names(SpData)=setLabels

# Number of data sets that we work
nSets = 2;

# Display the dimensions of the expression data (if you are confused by this construct, ignore it):
lapply(SpData, lapply, dim)


# Load module labels
x = load("PurpHed-SpeciesAnalysis-colorPurp-colorHed.RData")

# Create an object (list) holding the module labels for each set:
colorList=list(colorPurp,colorHed)

# Components of the list must be named so that the names can be matched to the names of SpData
names(colorList)=setLabels;
names(colorList)

                      

                                      ##################################
                               #  Calculation of module preservation statistics
                                      ##################################
# Note:

# The number of permutations drive the computation time
# of the module preservation function. Use at least 200 permutations
# Permutations are needed to calculate Zsummary and other permutation test statistics
# set random seed of permutation test analysis

# Setseed
set.seed(1)

#system.time( {mp = modulePreservation(SpData, colorList,referenceNetworks = c(1:2),loadPermutedStatistics = FALSE,nPermutations = 200,verbose = 3)} );

# Save the results of the module preservation analysis. 
 #save(mp, file = "PurpHed-PurpSpecific-modulePreservation.RData");
 load("PurpHed-PurpSpecific-modulePreservation.RData", .GlobalEnv)

```






```{r,WGNCA within species II}
# Work with I. purpurea and I. hederacea within competition only
softPower = 8;
adj_PurpAlone = adjacency(OTU_AlonePurp, power = softPower);
adj_PurpComp = adjacency(OTU_CompPurp, power = softPower);

# Turn adjacency into topological overlap
TOM_Comp = TOMsimilarity(adj_PurpComp);
dissTOM_Comp = 1-TOM_Comp

TOM_Alone = TOMsimilarity(adj_PurpAlone);
dissTOM_Alone = 1-TOM_Alone

# Call the hierarchical clustering function
Tree_Comp = hclust(as.dist(dissTOM_Comp), method = "average");
Tree_Alone = hclust(as.dist(dissTOM_Alone), method = "average");

colorh1b=as.character(cutreeStaticColor(Tree_Comp,cutHeight=.97,minSize=30))
table(colorh1b)

colorh2b=as.character(cutreeStaticColor(Tree_Alone,cutHeight=.97,minSize=30))
table(colorh2b)


# Set figure layout
layout(matrix(c(1:4), 2,2), heights = c(0.8, 0.2));

# Plot the dendrograms and module colors !!! TROUBLSHOOT PLOTS !!!

plotDendroAndColors(Tree_Comp,cbind(colorh1b, colorh2b),c("Inter (Ip)", "Inter (Ip)"),main="Competition (Ip)", dendroLabels=FALSE,setLayout = FALSE, abHeight = 0.97,marAll = c(0.3, 4.5, 2, 0.2))

plotDendroAndColors(Tree_Alone,cbind(colorh1b, colorh2b),c("Inter (Ip)", "Alone (Ip)"),main="Alone (Ip)", dendroLabels=FALSE,setLayout = FALSE, abHeight = 0.97)

#dev.off()

# Save the results

colorhALLb=as.character(colorh1b)
colorh2ALLb=as.character(colorh2b)

dynamicColors_Comp = labels2colors(colorh1b)
dynamicColors_Alone = labels2colors(colorh2b)

# Give the module colors more descriptive names
colorInter = colorhALLb;colorAlone = colorh2ALLb;


# Save module colors (assignments) and the indicator
save(colorInter, colorAlone, file = "InterAlone-TreatmentAnalysis-colorInter-colorAlone.RData")

# Calculate eigengenes
MEList_purp2 = moduleEigengenes(OTU_CompPurp, colors = dynamicColors_Comp)
MEs_purp2 = MEList_purp2$eigengenes

MEList_alone = moduleEigengenes(OTU_AlonePurp, colors = dynamicColors_Alone)
MEs_alone = MEList_alone$eigengene


# Calculate dissimilarity of module eigengenes
      #MEDiss = 1-cor(MEs);

# Cluster module eigengenes
      #METree = hclust(as.dist(MEDiss), method = "average");

# Plot the result
#sizeGrWindow(7, 6)
#plot(METree, main = "Clustering of module eigengenes",xlab = "", sub = "")
```

## Module Preservation analysis between treatment
```{r,WGNCA compare modules between treatment}
TrtData=list(IpInter = list(data = OTU_CompPurp), IpAlone = list(data = OTU_AlonePurp));

# Set labels
setLabels=c("Inter","Alone")
names(TrtData)=setLabels

# Number of data sets that we work
nSets = 2;

# Display the dimensions of the expression data (if you are confused by this construct, ignore it):
lapply(TrtData, lapply, dim)


# Load module labels
y = load("InterAlone-TreatmentAnalysis-colorInter-colorAlone.RData")

# Create an object (list) holding the module labels for each set:
colorList2=list(colorInter,colorAlone)

# Components of the list must be named so that the names can be matched to the names of SpData
names(colorList2)=setLabels;
names(colorList2)





                                      ##################################
                               #  Calculation of module preservation statistics
                                      ##################################
# Note:

# The number of permutations drive the computation time
# of the module preservation function. Use at least 200 permutations
# Permutations are needed to calculate Zsummary and other permutation test statistics
# set random seed of permutation test analysis

# Setseed
set.seed(1)

#system.time( {mp2 = modulePreservation(TrtData, colorList2,referenceNetworks = c(1:2),loadPermutedStatistics = FALSE,nPermutations = 200,verbose = 3)} );

# Save the results of the module preservation analysis. 
#save(mp2, file = "PurpComp-PurpAlone-modulePreservation.RData");
load(file="PurpComp-PurpAlone-modulePreservation.RData", .GlobalEnv)
```


## Preservation analysis statistics
```{r}
ref = 1
test = 2

statsObs = cbind(mp2$quality$observed[[ref]][[test]][, -1], mp2$preservation$observed[[ref]][[test]][, -1])
statsZ = cbind(mp2$quality$Z[[ref]][[test]][, -1], mp2$preservation$Z[[ref]][[test]][, -1]);

# Compare preservation to quality:
print( cbind(statsObs[, c("medianRank.pres", "medianRank.qual")],signif(statsZ[, c("Zsummary.pres", "Zsummary.qual")], 2)) )

# Module labels and module sizes are also contained in the results
modColors = rownames(mp2$preservation$observed[[ref]][[test]])
moduleSizes = mp2$preservation$Z[[ref]][[test]][, 1];
# leave grey and gold modules out
plotMods = !(modColors %in% c("grey"));

# Main titles for the plot
mains = c("Preservation Median rank", "Preservation Zsummary");



# If plotting into a file, close it

PresStats2=data.frame( cbind(statsObs[, c("medianRank.pres", "medianRank.qual")],signif(statsZ[, c("Zsummary.pres", "Zsummary.qual")], 2)))
PresStats2=PresStats2[plotMods,]
PresStats2$Module=row.names(PresStats2)
PresStats2$ModuleSize=moduleSizes[plotMods]

PresStats2[3,"Module"]="yellow"

PresStats2$Module=as.factor(PresStats2$Module)

MP_plot<-ggplot(PresStats2,aes(x=ModuleSize,y=medianRank.pres,color=Module))+
  geom_point(size=10)+
  scale_color_manual(values = c("blue", "brown","turquoise", "yellow"))+
  theme_bw()+
  ylab("Median Rank")+
  xlab("Module Size")+
  Tx

MP_plot2=ggplot(PresStats2,aes(x=ModuleSize,y=Zsummary.pres,color=Module))+
  geom_point(size=10,alpha=0.5)+
  scale_color_manual(values = c("blue", "brown","turquoise", "yellow"))+
  theme_bw()+
  ylab("Z-Summary Statistic")+
  xlab("Module Size")+
  Tx+
  geom_hline(aes(yintercept=0))+
  geom_hline(aes(yintercept=2),linetype = "dotdash",color="blue")+
  geom_hline(aes(yintercept=10),linetype = "dotdash",color="darkgreen")





# Auxiliary convenience variable
plotData = cbind(mp$preservation$observed[[ref]][[test]][, 2], mp$preservation$Z[[ref]][[test]][, 2])

# Main titles for the plot
mains = c("Preservation Median rank", "Preservation Zsummary");

modColors[which(modColors%in%"gold")]="yellow"
```


## Plot Phylum frequences per modules

```{r}


Taxa=tax_table(physeq_genus)
Taxa_phyla=data.frame("OTU"=row.names(Taxa),"Phylum"=Taxa[,2])


# I.purpurea alone
PhyModulesAlone=data.frame("OTU"=colnames(OTU_AlonePurp),"Modules"=dynamicColors_Alone)
PhyModulesAlone=merge(Taxa_phyla,PhyModulesAlone)

# I.purpurea competition
PhyModulesCompIp=data.frame("OTU"=colnames(OTU_CompPurp),"Modules"=dynamicColors_Comp)
PhyModulesCompIp=merge(Taxa_phyla,PhyModulesCompIp)

# I.hederacea competition
PhyModulesCompHed=data.frame("OTU"=colnames(OTU_CompHed),"Modules"=dynamicColors_H)
PhyModulesCompHed=merge(Taxa_phyla,PhyModulesCompHed)


Colors=c("gold","yellow","darkgreen","brown","orange","pink","red","darkblue","purple","#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#E6AB02", "#A6761D","#66C2A5", "#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F", "#E5C494","#FFFFE5", "#F7FCB9", "#D9F0A3",  "#004529","#FF7F00", "#CAB2D6", "#6A3D9A", "#FFFF99", "#B15928")



AlonePlotMod <- ggplot(PhyModulesAlone%>%filter(Phylum!="Bacteria_unclassified"), aes(Modules,group=Phylum,fill=Phylum)) + 
          geom_bar(aes(y = ((..count..)/sum(..count..)))) + 
          scale_y_continuous(labels=scales::percent) +
          #ylab("Relative Frequencies")+
          ylab("")+
          xlab("")+
          theme_bw()+ 
          scale_fill_manual(values = Colors)+
          theme(legend.position = "none")+
    #ggtitle(expression(paste(italic("I. purpurea"),"(alone)",sep="")))+
  #theme(plot.title=element_text(size=25,hjust=0),axis.text.x=element_text(size=25,hjust=0))+
    Tx




CompPlotMod_Hed <- ggplot(PhyModulesCompHed%>%filter(Phylum!="Bacteria_unclassified"), aes(Modules,group=Phylum,fill=Phylum)) + 
          geom_bar(aes(y = ((..count..)/sum(..count..)))) + 
          scale_y_continuous(labels=scales::percent) +
          #ylab("Relative Frequencies")+
            ylab("")+
          xlab("")+
          theme_bw()+ 
          scale_fill_manual(values = Colors)+
          Tx+
          theme(legend.position = "none")

  #ggtitle(expression(paste(italic("I. hederacea"),"(Competiton)",sep="")))



CompPlotMod_Purp <- ggplot(PhyModulesCompIp%>%filter(Phylum!="Bacteria_unclassified"), aes(Modules,group=Phylum,fill=Phylum)) + 
          geom_bar(aes(y = ((..count..)/sum(..count..)))) + 
          scale_y_continuous(labels=scales::percent) +
          #ylab("Relative Frequencies")+
            ylab("")+
          xlab("")+
          theme_bw()+ 
          scale_fill_manual(values = Colors)+
          Tx#+
          #theme(legend.position = "none")




## Save tables of phylum frequency per module per data set
Modules_CompIp<-PhyModulesCompIp%>%filter(Phylum!="Bacteria_unclassified")%>% # Filter unclassified
  group_by(Modules,Phylum)%>%
  dplyr::summarise(n = n()) %>%
  dplyr::mutate(freq = n / sum(n))#%>%# Uncomment next two lines to double check total sums to one
  #filter(Modules=="yellow")%>%  
  # summarise(TotalPercent=sum(freq))

## Save tables of phylum frequency per module per data set
Modules_AloneIp<-PhyModulesAlone%>%filter(Phylum!="Bacteria_unclassified")%>% # Filter unclassified
  group_by(Modules,Phylum)%>%
  dplyr::summarise(n = n()) %>%
  dplyr::mutate(freq = n / sum(n))#%>%# Uncomment next two lines to double check total sums to one
  #filter(Modules=="yellow")%>%  
  # summarise(TotalPercent=sum(freq))

## Save tables of phylum frequency per module per data set
Modules_CompHed<-PhyModulesCompHed%>%filter(Phylum!="Bacteria_unclassified")%>% # Filter unclassified
  group_by(Modules,Phylum)%>%
  dplyr::summarise(n = n()) %>%
  dplyr::mutate(freq = n / sum(n))#%>%# Uncomment next two lines to double check total sums to one
  #filter(Modules=="yellow")%>%  
  # summarise(TotalPercent=sum(freq))

Modules_CompHed$Group="Ihed-Comp"
Modules_CompIp$Group="Ip-Comp"  
Modules_AloneIp$Group="Ip-Alone"

totaldata=rbind(Modules_AloneIp,Modules_CompIp)

# Make a color dictionary so colors w phlyum are a constant across data
Phyla.All=data.frame("Phylum"=unique(totaldata$Phylum))
Phyla.All$Color=Colors[1:nrow(Phyla.All)]


Modules_CompIp<-merge(Phyla.All,Modules_CompIp)
Modules_AloneIp<-merge(Phyla.All,Modules_AloneIp)

Modules_CompIp$ModNum=""
Modules_CompIp[which(Modules_CompIp$Modules%in%"blue"),"ModNum"]="Module-1"
Modules_CompIp[which(Modules_CompIp$Modules%in%"brown"),"ModNum"]="Module-2"
Modules_CompIp[which(Modules_CompIp$Modules%in%"turquoise"),"ModNum"]="Module-3"
Modules_CompIp[which(Modules_CompIp$Modules%in%"yellow"),"ModNum"]="Module-4"

Modules_AloneIp$ModNum=""
Modules_AloneIp[which(Modules_AloneIp$Modules%in%"blue"),"ModNum"]="Module-1"
Modules_AloneIp[which(Modules_AloneIp$Modules%in%"brown"),"ModNum"]="Module-2"
Modules_AloneIp[which(Modules_AloneIp$Modules%in%"turquoise"),"ModNum"]="Module-3"
Modules_AloneIp[which(Modules_AloneIp$Modules%in%"yellow"),"ModNum"]="Module-4"


M_P_C<-ggplot(Modules_CompIp%>%filter(Phylum!="Bacteria_unclassified"),aes(x=ModNum,y=freq,fill=Phylum))+
  geom_bar(stat="identity")+
  ylab("")+
  xlab("")+
  theme_bw()+ 
  #theme(legend.position = "none")+
  scale_fill_manual(values = unique(Modules_CompIp$Color))+
  Tx+
  theme(axis.text.x = element_text(angle=45))

M_P_A<-ggplot(Modules_AloneIp%>%filter(Phylum!="Bacteria_unclassified"),aes(x=ModNum,y=freq,fill=Phylum))+
  geom_bar(stat="identity")+
  ylab("")+
  xlab("")+
  theme_bw()+ 
  theme(legend.position = "none")+
  scale_fill_manual(values = unique(Modules_AloneIp$Color))+
  Tx+
  theme(axis.text.x = element_text(angle=45))


# 1. Create the plots
#++++++++++++++++++++++++++++++++++
# Create a box plot with a top legend position

# 2. Save the legend
#+++++++++++++++++++++++
legend <- get_legend(M_P_C+theme(legend.position = "bottom"))
# 3. Remove the legend from the box plot
#+++++++++++++++++++++++
bp <- M_P_C + theme(legend.position="none")


A<-ggarrange(bp,M_P_A,labels=c("A","B"),nrow=1, font.label = list(size = 22, face = "plain"),label.x = 1)

ggarrange(A,legend,heights = c(1.5,1),nrow=2)
ggarrange(legend)


# Ven diagrams between modules and groups
# Prepare a palette of 3 colors with R colorbrewer:

myCol <- c("blue","brown","turquoise","yellow")


# Within I. purp alone and in competiton

# Blue module
PurpCompBlue=PhyModulesCompIp%>%
  filter(Modules=="blue")%>%
  select(Phylum)

PurpAloneBlue=PhyModulesAlone%>%
  filter(Modules=="blue")%>%
  select(Phylum)

#brown
PurpCompBrown=PhyModulesCompIp%>%
  filter(Modules=="brown")%>%
  select(Phylum)

PurpAloneBrown=PhyModulesAlone%>%
  filter(Modules=="brown")%>%
  select(Phylum)

# turquoise
PurpCompTurquoise=PhyModulesCompIp%>%
  filter(Modules=="turquoise")%>%
  select(Phylum)

PurpAloneTurquoise=PhyModulesAlone%>%
  filter(Modules=="turquoise")%>%
  select(Phylum)

#Yellow
PurpCompYellow=PhyModulesCompIp%>%
  filter(Modules=="yellow")%>%
  select(Phylum)

PurpAloneYellow=PhyModulesAlone%>%
  filter(Modules=="yellow")%>%
  select(Phylum)

# Within I. hed  in competiton

# Blue module
HedCompBlue=PhyModulesCompHed%>%
  filter(Modules=="blue")%>%
  select(Phylum)

HedCompTurquoise=PhyModulesCompHed%>%
  filter(Modules=="turquoise")%>%
  select(Phylum)


# Plot the Venn Diagrams

library(venn)
library(ggpolypath)

input<-lapply(list(PurpCompBlue$Phylum,PurpAloneBlue$Phylum),unique)

# Module 1
tmp <- venn::venn(input,snames = c("",""),zcolor = c("white","lightgrey"),opacity = 0.3,ilabels = TRUE,box = FALSE,ilcs = 2,sncs=2)
attr(tmp, "intersections")

A<-venn(input, ggplot = TRUE, linetype = "dashed",snames = c("Competition-Module-1","Alone-Module-1"),color=c("blue","grey"),alpha=0.5,box=FALSE,ilcs=29)+
  theme(text = element_text(size=30))


# Which are the bacteria phyla unique to the Brown Group?
Mod1Comp=unlist(input[1])
Mod1Alone=unlist(input[2])

length(Mod1Comp)
length(Mod1Alone)

Mod1Alone[-which(Mod1Alone%in%Mod1Comp)]

input2<-lapply(list(PurpCompBrown$Phylum,PurpAloneBrown$Phylum),unique)

tmp <- venn::venn(input2,snames =c("",""),zcolor = c("white","lightgrey"),opacity = 0.3,ilabels = TRUE,box = FALSE,ilcs = 2,sncs=2)
attr(tmp, "intersections")

A<-venn(input2, ggplot = TRUE, linetype = "dashed",snames = c("Competition-Module-2","Alone-Module-2"),color=c("blue","grey"),alpha=0.5,box=FALSE,ilcs=29)+
  theme(text = element_text(size=30))


# Which are the bacteria phyla unique to the Brown Group?
Mod2Comp=unlist(input2[1])
Mod2Alone=unlist(input2[2])

length(Mod2Comp)
length(Mod2Alone)

Mod2Comp[-which(Mod2Comp%in%Mod2Alone)]

input3<-lapply(list(PurpCompTurquoise$Phylum,PurpAloneTurquoise$Phylum),unique)

tmp <- venn::venn(input3,snames =c("",""),zcolor = c("white","lightgrey"),opacity = 0.3,ilabels = TRUE,box = FALSE,ilcs = 2,sncs=2)
attr(tmp, "intersections")

A<-venn(input3, ggplot = TRUE, linetype = "dashed",snames = c("Competition-Module-3","Alone-Module-3"),color=c("blue","grey"),alpha=0.5,box=FALSE,ilcs=29)+
  theme(text = element_text(size=30))


# Which are the bacteria phyla unique to the Brown Group?
Mod3Comp=unlist(input3[1])
Mod3Alone=unlist(input3[2])

length(Mod3Comp)
length(Mod3Alone)

Mod3Comp[-which(Mod3Comp%in%Mod3Alone)]
Mod3Alone[-which(Mod3Alone%in%Mod3Comp)]

input4<-lapply(list(PurpCompYellow$Phylum,PurpAloneYellow$Phylum),unique)

tmp <- venn::venn(input4,snames =c("",""),zcolor = c("white","lightgrey"),opacity = 0.3,ilabels = TRUE,box = FALSE,ilcs = 2,sncs=2)
attr(tmp, "intersections")

A<-venn(input4, ggplot = TRUE, linetype = "dashed",snames = c("Competition-Module-4","Alone-Module-4"),color=c("blue","grey"),alpha=0.5,box=FALSE,ilcs=29)+
  theme(text = element_text(size=30))


# Which are the bacteria phyla unique to the Brown Group?
Mod4Comp=unlist(input4[1])
Mod4Alone=unlist(input4[2])

Mod4Comp[-which(Mod4Comp%in%Mod4Alone)]
Mod4Alone[-which(Mod4Alone%in%Mod4Comp)]

# I. purpurea alone

input2<-lapply(list(PurpAloneBlue$Phylum,PurpAloneBrown$Phylum,PurpAloneTurquoise$Phylum,PurpAloneYellow$Phylum),unique)

tmp <- venn::venn(input2,snames = c("Blue","Brown","Turquoise","Yellow"),zcolor = c("blue","brown","turquoise","yellow"),opacity = 0.3,ilabels = TRUE,box = FALSE,ilcs = 1)
attr(tmp, "intersections")

B<-venn(input2, ggplot = TRUE, linetype = "dashed",snames = c("Blue","Brown","Turquoise","Yellow"),color=c("blue","brown","turquoise","yellow"),alpha=0.5,box=FALSE)+
  theme(text = element_text(size=22))

# Which are the bacteria phyla unique to the Brown Group?
Brown2=unlist(input2[2])
AllElse2=unlist(input2[-2])

length(Brown2)
length(AllElse2)

UniqBrown2=Brown2[!Brown2%in%AllElse2] # Select bacteria unique to the Brown Group

dput(droplevels(UniqBrown2))

# I. hederacea in competition

input3<-lapply(list(HedCompBlue$Phylum,HedCompTurquoise$Phylum),unique)

tmp <- venn::venn(input3,snames = c("Blue","Turquoise"),zcolor = c("blue","turquoise"),opacity = 0.3,ilabels = TRUE,box = FALSE,ilcs = 1)
attr(tmp, "intersections")

B<-venn(input3, ggplot = TRUE, linetype = "dashed",snames = c("Blue","Turquoise"),color=c("blue","turquoise","yellow"),alpha=0.5,box=FALSE)+
  theme(text = element_text(size=22))

# Which are the bacteria phyla unique to the Yewllow Group?
Blue=unlist(input3[1])
Turq=unlist(input3[-1])

length(Blue)
length(Turq)

UniqBlue=Blue[!Blue%in%Turq] # Select bacteria unique to the Blue Group
UniqTurq=Turq[!Turq%in%Blue] # Select bacteria unique to the Turq Group

dput(droplevels(UniqBlue))
dput(droplevels(UniqTurq))



```




## Hypothesis testing WGCNA

```{r,WGNCA within species III}

# Data for I. purpurea in competition 
MEs_data_Comp=cbind(MEs_P,SAMPLE[which(SAMPLE$Sample_ID%in%row.names(MEs_P)),c("TRT","ML","Block","Combos")])

# Data for I. purpurea ALONE
MEs_data_Alone=cbind(MEs_alone,SAMPLE[which(SAMPLE$Sample_ID%in%row.names(MEs_alone)),c("TRT","ML","Block","Combos")])


# All data
Combined=rbind(MEs_alone,MEs_P)
MEs_data=cbind(Combined,SAMPLE[which(SAMPLE$Sample_ID%in%row.names(Combined)),c("TRT","ML","Block","Combos")])


# Linear mixed model
lmmMod<-lmer(MEblue~TRT+Block+(1|ML),MEs_data)
anova(lmmMod)
ranova(lmmMod)


lmmMod<-lmer(MEbrown~TRT+Block+(1|ML),MEs_data)
anova(lmmMod)
ranova(lmmMod)

lmmMod<-lmer(MEturquoise~TRT+Block+(1|ML),MEs_data)
anova(lmmMod)
ranova(lmmMod)


lmmMod<-lmer(MEyellow~TRT+Block+(1|ML),MEs_data)
anova(lmmMod)
ranova(lmmMod)

RootAveraged$TRT=RootAveraged$Trt

## Add fitness and root data with the eigen value data

MEs_data_Alone=merge(MEs_data_Alone,RootAveraged)
MEs_data_Alone=merge(MEs_data_Alone,FitAveraged)
head(MEs_data_Alone)

MEs_data_Comp=merge(MEs_data_Comp,RootAveraged)
MEs_data_Comp=merge(MEs_data_Comp,FitAveraged)
head(MEs_data_Comp)

# Rename root columns
colnames(MEs_data_Comp)[grep("PC1|PC2|PC3|PC4",names(MEs_data_Comp))]=c("RootTopology","RootArchitecture","RootSize","RootMorphology")
colnames(MEs_data_Alone)[grep("PC1|PC2|PC3|PC4",names(MEs_data_Alone))]=c("RootTopology","RootArchitecture","RootSize","RootMorphology")

# Rename modules
colnames(MEs_data_Comp)[grep("blue|brown|turquoise|yellow",names(MEs_data_Comp))]=c("Module-1","Module-2","Module-3","Module-4")
colnames(MEs_data_Alone)[grep("blue|brown|turquoise|yellow",names(MEs_data_Alone))]=c("Module-1","Module-2","Module-3","Module-4")

                                                        #############################################
                                                         # Let's exaamine correlations with modules
                                                        #############################################
# Index columns of interest
I=grep("Module|Root|Relative",names(MEs_data_Alone))
  Corr<-rcorr(as.matrix(MEs_data_Alone[c(I)]))
  corrplot(Corr$r,type="upper")

  
  
# Display significance
# Mark the insignificant coefficients according to the specified p-value significance level

col <- colorRampPalette(c("#BB4444", "#EE9988", "#FFFFFF", "#77AADD", "#4477AA"))
C1=corrplot(Corr$r, type = "upper",
         p.mat = Corr$P, sig.level = 0.05, tl.col = "black", col=col(200))

C1=ggcorrplot(Corr$r, type="lower",p.mat=Corr$P)



J=grep("Module|Root|Relative",names(MEs_data_Comp))
  Corr2<-rcorr(as.matrix(MEs_data_Comp[c(J)]))
  corrplot(Corr$r,type="upper")

# Display significance
# Mark the insignificant coefficients according to the specified p-value significance level

C2=corrplot(Corr2$r, type = "upper",
         p.mat = Corr2$P, sig.level = 0.05, tl.col = "black", col=col(200))

C2=ggcorrplot(Corr2$r, type="lower",p.mat=Corr2$P)

ggarrange(C1,C2,common.legend = T,nrow=1,labels = "AUTO")


# Alternative correlation display--bar graphs
library(ggstatsplot)

# Create data frames to correct structure for making bar graphs

# (1) create correlation data frames with root traits
# (2) for ea treatment
# (3) Repeat to add P-values

# R-square values

  # Alone treatment
  AloneCorr=data.frame(Corr$r)
  AloneCorr$RootTrait=row.names(AloneCorr)
  AloneCorr<-AloneCorr[grep("Root",AloneCorr$RootTrait),]
  AloneCorr<-AloneCorr[,grep("Module|RootTrait",names(AloneCorr))]
  AloneCorr$Trt="Alone"
  
  # Competition treatment
  CompCorr=data.frame(Corr2$r)
  CompCorr$RootTrait=row.names(CompCorr)
  CompCorr<-CompCorr[grep("Root",CompCorr$RootTrait),]
  CompCorr<-CompCorr[,grep("Module|RootTrait",names(CompCorr))]
  CompCorr$Trt="Comp"
  
  RootCorr=rbind(CompCorr,AloneCorr)
  
# Reshape to long format
Rcorr=  melt(RootCorr, id.vars=c("RootTrait","Trt"),variable.name = "Module",value.name = "r2")

# P-values


  # Alone treatment
  AlonepVal=data.frame(Corr$P)
  AlonepVal$RootTrait=row.names(AlonepVal)
  AlonepVal<-AlonepVal[grep("Root",AlonepVal$RootTrait),]
  AlonepVal<-AlonepVal[,grep("Module|RootTrait",names(AlonepVal))]
  AlonepVal$Trt="Alone"
  
  # Competition treatment
  ComppVal=data.frame(Corr2$P)
  ComppVal$RootTrait=row.names(ComppVal)
  ComppVal<-ComppVal[grep("Root",ComppVal$RootTrait),]
  ComppVal<-ComppVal[,grep("Module|RootTrait",names(ComppVal))]
  ComppVal$Trt="Comp"
  
  RootpVal=rbind(ComppVal,AlonepVal)
  
# Reshape to long format
RpVal  =melt(RootpVal, id.vars=c("RootTrait","Trt"),variable.name = "Module",value.name = "pvalue")

# combe R-squared and P-values
CorrPvalRoots=merge(Rcorr,RpVal)

                  # Plot Root traits and r-squared, add an asterick to significant correlations

# Plot parameters for y axis min and max
ymin=min(CorrPvalRoots$r2)
ymax=max(CorrPvalRoots$r2)

# Root topology and module correlations

ArchMod=ggplot(CorrPvalRoots%>%filter(RootTrait=="RootArchitecture"),aes(Module,r2,fill=Trt))+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  scale_fill_manual(values = c("black","grey"))+
  #geom_text(aes(label = ifelse(pvalue < 0.05, "*", "")), hjust=1.5, color="black", size=5)+
  theme_minimal()+
  xlab("")+
  ylab("")+
  ggtitle("")+
  ylim(ymin,ymax)+
  labs()+
  labs(fill="Treatment")+
  scale_x_discrete(labels= c("Module 1","Module 2", "Module 3", "Module 4"))+
  theme(axis.text.x = element_text(angle=45),axis.text=element_text(color="black",size=15,vjust = 0.5,hjust=1))

TopMod=ggplot(CorrPvalRoots%>%filter(RootTrait=="RootTopology"),aes(Module,r2,fill=Trt))+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  scale_fill_manual(values = c("black","grey"))+
  #geom_text(aes(label = ifelse(pvalue < 0.05, "*", "")), hjust=1.5, color="black", size=5)+
  theme_minimal()+
  xlab("")+
  ylab("")+
  #ggtitle("Root Topology")+
  ylim(ymin,ymax)+
  labs(fill="Treatment")+
  scale_x_discrete(labels= c("Module 1","Module 2", "Module 3", "Module 4"))+
  theme(axis.text.x = element_text(angle=45),axis.text=element_text(color="black",size=15,vjust = 0.5,hjust=1))

MorphMod=ggplot(CorrPvalRoots%>%filter(RootTrait=="RootMorphology"),aes(Module,r2,fill=Trt))+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  scale_fill_manual(values = c("black","grey"))+
  #geom_text(aes(label = ifelse(pvalue < 0.05, "*", "")), hjust=2,color="black", size=5)+
  theme_minimal()+
  xlab("")+
  ylab("")+
  ggtitle("")+
  ylim(ymin,ymax)+
  labs(fill="Treatment")+
  scale_x_discrete(labels= c("Module 1","Module 2", "Module 3", "Module 4"))+
  theme(axis.text.x = element_text(angle=45),axis.text=element_text(color="black",size=15,vjust = 0.5,hjust=1))

SizeMod=ggplot(CorrPvalRoots%>%filter(RootTrait=="RootSize"),aes(Module,r2,fill=Trt))+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  scale_fill_manual(values = c("black","grey"))+
  theme_minimal()+
  xlab("")+
  ylab("")+
  #ggtitle("Root Size")+
  ylim(ymin,ymax)+
  labs(fill="Treatment")+
  scale_x_discrete(labels= c("Module 1","Module 2", "Module 3", "Module 4"))+
  theme(axis.text.x = element_text(angle=45),axis.text=element_text(color="black",size=15,vjust = 0.5,hjust=1))

ggarrange(TopMod,ArchMod,SizeMod,MorphMod,nrow=2,ncol=2,common.legend = T,labels="AUTO",align=c("hv"),label.x = 0.20,font.label = list(size=20))



# Repeat for fitness and module correlations

# (1) create correlation data frames with fitness
# (2) for ea treatment
# (3) Repeat to add P-values

# R-square values

  # Alone treatment
  AloneCorr=data.frame(Corr$r)
  AloneCorr$Trait=row.names(AloneCorr)
  AloneCorr<-AloneCorr[grep("Fit",AloneCorr$Trait),]
  AloneCorr<-AloneCorr[,grep("Module",names(AloneCorr))]
  AloneCorr$Trt="Alone"
  
  # Competition treatment
  CompCorr=data.frame(Corr2$r)
  CompCorr$Trait=row.names(CompCorr)
  CompCorr<-CompCorr[grep("Fit",CompCorr$Trait),]
  CompCorr<-CompCorr[,grep("Module",names(CompCorr))]
  CompCorr$Trt="Comp"
  
  FitCorr=rbind(CompCorr,AloneCorr)
  
# Reshape to long format
RcorrFit=  melt(FitCorr, id.vars=c("Trt"),variable.name = "Module",value.name = "r2")

# P-values

  # Alone treatment
  AlonepVal=data.frame(Corr$P)
  AlonepVal$Trait=row.names(AlonepVal)
  AlonepVal<-AlonepVal[grep("Fit",AlonepVal$Trait),]
  AlonepVal<-AlonepVal[,grep("Module",names(AlonepVal))]
  AlonepVal$Trt="Alone"
  
  # Competition treatment
  ComppVal=data.frame(Corr2$P)
  ComppVal$Trait=row.names(ComppVal)
  ComppVal<-ComppVal[grep("Fit",ComppVal$Trait),]
  ComppVal<-ComppVal[,grep("Module",names(ComppVal))]
  ComppVal$Trt="Comp"
  
  Fitpval=rbind(ComppVal,AlonepVal)
  
# Reshape to long format
pvalFit=  melt(Fitpval, id.vars=c("Trt"),variable.name = "Module",value.name = "pvalue")


# combe R-squared and P-values
CorrPvalFit=merge(RcorrFit,pvalFit)

                  # Plot Root traits and r-squared, add an asterick to significant correlations

# Plot parameters for y axis min and max
ymin=min(CorrPvalFit$r2)
ymax=max(CorrPvalFit$r2)

# Root topology and module correlations

FitMod=ggplot(CorrPvalFit,aes(Module,r2,fill=Trt))+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  scale_fill_manual(values = c("black","grey"))+
 # geom_text(aes(label = ifelse(pvalue < 0.055, "*", "")),hjust=-3,color="black", size=5)+
  theme_bw()+
  xlab("")+
  ylab("")+
  ggtitle("")+
  ylim(ymin,ymax)+
  theme(legend.position = "top")+
  labs(fill="Treatment")+
  scale_x_discrete(labels= c("Module 1","Module 2", "Module 3", "Module 4"))+
  theme(axis.text.x = element_text(angle=45),axis.text=element_text(color="black",size=15,vjust = 0.5,hjust=1))
  

```



# Network Plotting

## I. purpurea

```{r}

# Setting up a color dictionary to color by Phylum

Modules_CompIp1<-PhyModulesCompIp%>%filter(Phylum!="Bacteria_unclassified")%>% # Filter unclassified
  group_by(Phylum)%>%
  summarise(n = n()) %>%
  mutate(freq = n / sum(n))

ColorTop=Modules_CompIp1%>% 
  #filter(Modules=="blue")%>%
  arrange_(~ desc(freq)) %>%
  group_by_(~Phylum)%>%
  head(n = 5)

ColorList=c('#d73027','#f46d43','#fdae61','#fee090','#e0f3f8','#abd9e9','#74add1','#4575b4') 

ColorKey=data.frame("Phylum"=unique(ColorTop$Phylum),"Color"=ColorList[1:length(unique(ColorTop$Phylum))])


Colors=rep(times=620,x="grey")

for(i in 1:nrow(PhyModulesCompIp)){
  if(PhyModulesCompIp[i,"Phylum"]%in%ColorTop$Phylum){
  Colors[i]= paste(ColorKey[which(ColorKey$Phylum%in%PhyModulesCompIp[i,"Phylum"]),"Color"]) 
  }  
  }

# Plot networks
set.seed(123)


#                           Color by Modules

# I. purpurea ALONE
adj <- TOM_Alone
adj[adj > 0.01] = 1
adj[adj != 1] = 0
network <- graph.adjacency(adj,weighted = T)
network <- simplify(network)  # removes self-loops
V(network)$color <- dynamicColors_Alone # Coloring by modules
par(mar=c(0,0,0,0))

# remove unconnected nodes
network <- igraph::delete.vertices(network, igraph::degree(network)==0)
NiceLay=layout_nicely(network,3)

#plot(network, layout=layout.fruchterman.reingold(network), edge.arrow.size = 0.5,vertex.label=NA)

plot(network, layout=NiceLay, edge.arrow.size = 0.5,vertex.label=NA)






#                                           REPEAT for I.purp in Competition
#                           Color by Modules

# I. purpurea in competition
adj <- TOM_Comp
adj[adj > 0.01] = 1
adj[adj != 1] = 0
network_comp <- graph.adjacency(adj,weighted = T)
network_comp <- simplify(network_comp)  # removes self-loops
V(network_comp)$color <- dynamicColors_Comp # Coloring by modules
par(mar=c(0,0,0,0))


# remove unconnected nodes
network_comp <- igraph::delete.vertices(network_comp, igraph::degree(network_comp)==0)
NiceLay=layout_nicely(network_comp,2)
CircleLay=layout.circle(network_comp)

plot(network_comp, layout=layout.fruchterman.reingold(network_comp), edge.arrow.size = 0.5,vertex.label=NA)

# plot(network_comp, layout=CircleLay, edge.arrow.size = 0.5,vertex.label=NA)

#                           Color by Phylum


#V(network_comp)$color <- Colors
par(mar=c(0,0,0,0))
# remove unconnected nodes
network_comp <- igraph::delete.vertices(network_comp, degree(network)==0)
plot(network_comp, layout=layout.fruchterman.reingold(network), edge.arrow.size = 0.5,vertex.label=NA)


```




## I. hederacea
```{r}
# First lets indicate top 5 phyla according to Rel Freq. (we will color these in the plots)

## Save tables of phylum frequency per module per data set


  CountPhyTop=function(x){
    
Output<-PhyModulesCompHed%>%filter(Phylum!="Bacteria_unclassified",Modules==x)%>% # Filter unclassified
  group_by(Phylum)%>%
  summarise(n = n()) %>%
  mutate(freq = n / sum(n))%>%
  arrange_(~ desc(freq))
 return(Output)
}

ResultsCount=lapply(unique(PhyModulesCompHed$Modules),CountPhyTop)

# Below code needs to be fixed!

ColorTop=Modules_Hed1%>% 
  #filter(Modules=="blue")%>%
  arrange_(~ desc(freq)) %>%
  group_by_(~Modules)%>%
  head(n = 5)

ColorList=c('#d73027','#f46d43','#fdae61','#fee090','#e0f3f8','#abd9e9','#74add1','#4575b4') 

ColorKey=data.frame("Phylum"=unique(ColorTop$Phylum),"Color"=ColorList[1:length(unique(ColorTop$Phylum))])


Colors=rep(times=620,x="grey")

for(i in 1:nrow(PhyModulesCompHed)){
  if(PhyModulesCompHed[i,"Phylum"]%in%ColorTop$Phylum){
  Colors[i]= paste(ColorKey[which(ColorKey$Phylum%in%PhyModulesCompHed[i,"Phylum"]),"Color"]) 
  }  
  }


### Plot the modules of I. hederacea
set.seed(123)
# I. hederacea in competition
adj <- TOM_H
adj[adj > 0.001] = 1
adj[adj != 1] = 0
networkH <- graph.adjacency(adj)
networkH <- simplify(networkH)  # removes self-loops
V(networkH)$color <- dynamicColors_H


 #layout=layout.fruchterman.reingold(networkH)
par(mar=c(0,0,0,0))

# remove unconnected nodes
networkH <- igraph::delete.vertices(networkH, igraph::degree(networkH)==0)


Layout=layout_nicely(networkH,dim = 3)

plot(networkH,layout=layout.fruchterman.reingold(networkH), edge.arrow.size = 0.3,vertex.label=NA,rescale=T)
 
plot(networkH,layout=Layout, edge.arrow.size = 0.3,vertex.label=NA,rescale=T)

plot.new()
dev.off()


```



## Characterize module networks

1. Connectivity within modules
2. Centrality within modules
3. ID hub bacteria

```{r}
# Set phylum as network attribute
network<-igraph::set_vertex_attr(network,"Phylum",index=igraph::V(network),PhyModulesAlone$Phylum)

network_comp<-igraph::set_vertex_attr(network_comp,"Phylum",index=igraph::V(network_comp),PhyModulesCompIp$Phylum)


# Subset by color and examine characteristics of these 'sub' networks
Modules=unique(V(network)$color)

NetworkStats=data.frame("Modules"=unique(V(network)$color),"ClustCoef"=0,"Centralization"=0,"Density"=0,"AvgNeighborNum"=0,"Node_n"=0)


# Run a for loop to fill in table with network characteristics

for(i in 1:length(Modules)){
  SubNet<- igraph::delete.vertices(network,V(network)[color != Modules[i]]) # Subset the network by module
  
  # Save the network stats as objects
  
  ClustCoef=transitivity(SubNet)
  Central=centr_eigen(SubNet)$centralization
  Dens=edge_density(SubNet,loop=F)
  AvNeighbor=mean(knn(SubNet)$knn,na.rm=T)
  NodeN=gorder(SubNet)
  
  # Add stats to data frame
  NetworkStats[i,"ClustCoef"]=round(ClustCoef,2)
  NetworkStats[i,"Centralization"]=round(Central,2)
  NetworkStats[i,"Density"]=round(Dens,2)
  NetworkStats[i,"AvgNeighborNum"]=round(AvNeighbor,2)
  NetworkStats[i,"Node_n"]=round(NodeN,2)
}


BlueAlone<- igraph::delete.vertices(network,V(network)[color != Modules[4]])
BlueAlone<-simplify(BlueAlone, remove.multiple = F, remove.loops = T)

BAlone=plot(BlueAlone,vertex.label=NA,edge.arrow.size=0.1)

TurqAlone<- igraph::delete.vertices(network,V(network)[color != Modules[1]])
TurqAlone<-simplify(TurqAlone, remove.multiple = F, remove.loops = T) # Simplify network
TAlone=plot(TurqAlone,vertex.label=NA,edge.arrow.size=0.1,vertex.color=paste(Modules[1]))


YellAlone<- igraph::delete.vertices(network,V(network)[color != Modules[2]])
YellAlone<-simplify(YellAlone, remove.multiple = F, remove.loops = T) # Simplify network
YAlone=plot(YellAlone,vertex.label=NA,edge.arrow.size=0.1,vertex.color=paste(Modules[2]))

BrAlone<- igraph::delete.vertices(network,V(network)[color != Modules[3]])
BrAlone<-simplify(BrAlone, remove.multiple = F, remove.loops = T) # Simplify network
Bralone=plot(BrAlone,vertex.label=NA,edge.arrow.size=0.1,vertex.color=paste(Modules[3]))


BlueComp<- igraph::delete.vertices(network_comp,V(network_comp)[color != Modules[4]])
BlueComp<-simplify(BlueComp, remove.multiple = F, remove.loops = T) # Simplify network
Bcomp=plot(BlueComp,vertex.label=NA,edge.arrow.size=0.1)

TurqComp<- igraph::delete.vertices(network_comp,V(network_comp)[color != Modules[1]])
TurqComp<-simplify(TurqComp, remove.multiple = F, remove.loops = T) # Simplify network
Tcomp=plot(TurqComp,vertex.label=NA,edge.arrow.size=0.1,vertex.color=paste(Modules[1]))

YellComp<- igraph::delete.vertices(network_comp,V(network_comp)[color != Modules[2]])
YellComp<-simplify(YellComp, remove.multiple = F, remove.loops = T) # Simplify network

Ycomp=plot(YellComp,vertex.label=NA,edge.arrow.size=0.1,vertex.color=paste(Modules[2]))

BrComp<- igraph::delete.vertices(network_comp,V(network_comp)[color != Modules[3]])
BrComp<-simplify(BrComp, remove.multiple = F, remove.loops = T) # Simplify network

Brcomp=plot(BrComp,vertex.label=NA,edge.arrow.size=0.1,vertex.color=paste(Modules[3]))


                                  # Repeat for I. purp in competition

# Subset by color and examine characteristics of these 'sub' network_comps
ModulesComp=unique(V(network_comp)$color)

network_compStats=data.frame("Modules"=unique(V(network_comp)$color),"ClustCoef"=0,"Centralization"=0,"Density"=0,"AvgNeighborNum"=0,"Node_n"=0)


# Run a for loop to fill in table with network_comp characteristics

for(i in 1:length(ModulesComp)){
  SubNet<- igraph::delete.vertices(network_comp,V(network_comp)[color != ModulesComp[i]]) # Subset the network_comp by module
  
  # Save the network_comp stats as objects
  
  ClustCoef=transitivity(SubNet)
  Central=centr_eigen(SubNet)$centralization
  Dens=edge_density(SubNet,loop=F)
  AvNeighbor=mean(knn(SubNet)$knn,na.rm=T)
  NodeN=gorder(SubNet)
  
  # Add stats to data frame
  network_compStats[i,"ClustCoef"]=round(ClustCoef,2)
  network_compStats[i,"Centralization"]=round(Central,2)
  network_compStats[i,"Density"]=round(Dens,2)
  network_compStats[i,"AvgNeighborNum"]=round(AvNeighbor,2)
  network_compStats[i,"Node_n"]=round(NodeN,2)
}

# Centrality 
closeness.cent <- igraph::closeness(SubNet, mode="all")
Top=max(closeness.cent)
Index=which(closeness.cent==Top)


                                  # Repeat for I. hed in competition

# Subset by color and examine characteristics of these 'sub' networkHs
ModulesH=unique(V(networkH)$color)

networkHStats=data.frame("Modules"=unique(V(networkH)$color),"ClustCoef"=0,"Centralization"=0,"Density"=0,"AvgNeighborNum"=0,"Node_n"=0)


# Run a for loop to fill in table with networkH characteristics

for(i in 1:length(ModulesH)){
  SubNet<- igraph::delete.vertices(networkH,V(networkH)[color != ModulesH[i]]) # Subset the networkH by module
  
  # Save the networkH stats as objects
  
  ClustCoef=transitivity(SubNet)
  Central=centr_eigen(SubNet)$centralization
  Dens=edge_density(SubNet,loop=F)
  AvNeighbor=mean(knn(SubNet)$knn,na.rm=T)
  NodeN=gorder(SubNet)
  
  # Add stats to data frame
  networkHStats[i,"ClustCoef"]=round(ClustCoef,2)
  networkHStats[i,"Centralization"]=round(Central,2)
  networkHStats[i,"Density"]=round(Dens,2)
  networkHStats[i,"AvgNeighborNum"]=round(AvNeighbor,2)
  networkHStats[i,"Node_n"]=round(NodeN,2)
}




NetworkStats$Data="I.purpurea (Alone)"
NetworkStats$Trt="Alone"

network_compStats$Data="I.purpurea (Comp)"
network_compStats$Trt="Competition"


NetworkStatsAll=rbind(NetworkStats,network_compStats)
NetworkStatsAll$Modules=as.character(NetworkStatsAll$Modules)

# Rename modules
blue=grep("blue",NetworkStatsAll$Modules)
NetworkStatsAll[blue,"Modules"]="Module-1"


brown=grep("brown",NetworkStatsAll$Modules)
NetworkStatsAll[brown,"Modules"]="Module-2"

turq=grep("turquoise",NetworkStatsAll$Modules)
NetworkStatsAll[turq,"Modules"]="Module-3"

yellow=grep("yellow",NetworkStatsAll$Modules)
NetworkStatsAll[yellow,"Modules"]="Module-4"

library(DT)
datatable(NetworkStatsAll)


## Runing functions to produce multipe plots
 
# Aesthetics
Tx2<-theme(axis.text.y = element_text(size=20),
          axis.title.y = element_text(size=25))+
          theme(axis.text.x = element_text(vjust = 1, hjust=1,size=20),
          axis.title.x = element_text(angle=0,size=25),
          plot.title=element_text(size=15,hjust=0))



# Clustering Coefficient

P1=ggplot(NetworkStatsAll,aes(Modules,ClustCoef,fill=Trt),color="black")+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  theme_bw()+
  scale_fill_manual(values=c("black","grey"))+
  Tx+
  theme(axis.text.x = element_text(angle=45))+
  ylab("Clustering Coefficient")+
  Tx2+
  xlab("")


# Centralization

P2=ggplot(NetworkStatsAll,aes(Modules,Centralization,fill=Trt),color="black")+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  theme_bw()+
  scale_fill_manual(values=c("black","grey"))+
  Tx+
  theme(axis.text.x = element_text(angle=45))+
  ylab("Centralization")+
  Tx2+
  xlab("")


# Density
P3=ggplot(NetworkStatsAll,aes(Modules,Density,fill=Trt),color="black")+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  theme_bw()+
  scale_fill_manual(values=c("black","grey"))+
  Tx+
  theme(axis.text.x = element_text(angle=45),legend.position = "top")+
  ylab("Density")+
  Tx2+
  xlab("")+
  labs(fill="Treatment")+
  scale_x_discrete(labels= c("Module 1","Module 2", "Module 3", "Module 4"))
# Average Neighbor
P4=ggplot(NetworkStatsAll,aes(Modules,AvgNeighborNum,fill=Trt),color="black")+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  theme_bw()+
  scale_fill_manual(values=c("black","grey"))+
  Tx+
  theme(axis.text.x = element_text(angle=45))+
  ylab("Average Neighbor")+
  Tx2+
  xlab("")


# Node number

P5=ggplot(NetworkStatsAll,aes(Modules,Node_n,fill=Trt),color="black")+
  geom_bar(stat="identity",position="dodge",alpha=0.5)+
  theme_bw()+
  scale_fill_manual(values=c("black","grey"))+
  Tx+
  theme(axis.text.x = element_text(angle=45))+
  ylab("Node Number")+
  Tx2+
  xlab("")

ggarrange(P2,P3,ncol=2,common.legend = T)

Labels=c("Cluster Coefficient","Centralization","Density","Average Neighbor","Node Number")

Stats=names(NetworkStats)[2:6]


# Multiple plots for I. purp in competition


PlotFunNet=function(x){
  Match=substr(paste(x), 1, 2)
  Ylabel=Labels[grep(Match,Labels)]
  colors=as.vector(network_compStats$Modules[order(network_compStats$Modules)])
  Plot=ggplot(network_compStats,aes(Modules,network_compStats[,x],fill=Modules))+
  geom_bar(stat="identity",alpha=0.5)+
  theme_bw()+
  scale_fill_manual(values=c(colors))+
  Tx2+
  theme(axis.text.x = element_text(angle=45))+
  #facet_grid(~Data)+
  ylab("")+
  xlab("")+
  theme(legend.position = "none")+
  ggtitle(Ylabel)

  return(Plot)
}

PlotsListed=lapply(Stats,PlotFunNet)

PurpComp=do.call(ggpubr::ggarrange, c(PlotsListed, list(ncol=5)))



# Multiple plots for I. purp in alone

PlotFunNet=function(x){
  Match=substr(paste(x), 1, 2)
  Ylabel=Labels[grep(Match,Labels)]
  colors=as.vector(NetworkStats$Modules[order(NetworkStats$Modules)])
  Plot=ggplot(NetworkStats,aes(Modules,NetworkStats[,x],fill=Modules))+
  geom_bar(stat="identity",alpha=0.5)+
  theme_bw()+
  scale_fill_manual(values=c(colors))+
  Tx2+
  theme(axis.text.x = element_text(angle=45))+
  #facet_grid(~Data)+
  ylab("")+
  xlab("")+
  #ggtitle(Ylabel)+
  theme(legend.position = "none")
  return(Plot)
}

PlotsListed=lapply(Stats,PlotFunNet)

PurpAlone=do.call(ggpubr::ggarrange, c(PlotsListed, list(ncol=5)))

ggarrange(PurpComp,PurpAlone,nrow=2)
```







